<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Sclerostin pQTL MR - 2&nbsp; Data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./03_ld_matrix.html" rel="next">
<link href="./index.html" rel="prev">
<link href="././images/geni.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="site_libs/core-js-2.5.3/shim.min.js"></script>
<script src="site_libs/react-18.2.0/react.min.js"></script>
<script src="site_libs/react-18.2.0/react-dom.min.js"></script>
<script src="site_libs/reactwidget-1.0.0/react-tools.js"></script>
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/reactable-0.4.4/reactable.css" rel="stylesheet">
<script src="site_libs/reactable-binding-0.4.4/reactable.js"></script>
<script src="site_libs/font-awesome-5.13.0/js/script.js"></script>
<link rel="shortcut icon" href="images/geni_favicon.png">

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./02_data.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Data</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-center sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="././images/geni.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Sclerostin pQTL MR</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/jrs95/sclerostin_pqtl_mr" rel="" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02_data.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Data</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03_ld_matrix.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">LD matrix</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04_sost_assoc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title"><em>SOST</em> region associations</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05_pqtl_assoc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Sclerostin pQTL associations</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./06_mr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Mendelian randomization</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./07_coloc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Colocalization analyses</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./08_sensitivity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Sensitivity analyses</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./09_references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./10_ackowledgements.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Acknowledgements</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./11_acroynms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Acronyms</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#datasets" id="toc-datasets" class="nav-link active" data-scroll-target="#datasets"><span class="header-section-number">2.1</span> Datasets</a>
  <ul class="collapse">
  <li><a href="#sclerostin-pqtls" id="toc-sclerostin-pqtls" class="nav-link" data-scroll-target="#sclerostin-pqtls"><span class="header-section-number">2.1.1</span> Sclerostin pQTLs</a></li>
  <li><a href="#study-information" id="toc-study-information" class="nav-link" data-scroll-target="#study-information"><span class="header-section-number">2.1.2</span> Study information</a></li>
  </ul></li>
  <li><a href="#data-processing" id="toc-data-processing" class="nav-link" data-scroll-target="#data-processing"><span class="header-section-number">2.2</span> Data processing</a>
  <ul class="collapse">
  <li><a href="#ld-matrix" id="toc-ld-matrix" class="nav-link" data-scroll-target="#ld-matrix"><span class="header-section-number">2.2.1</span> LD matrix</a></li>
  <li><a href="#gwas-data" id="toc-gwas-data" class="nav-link" data-scroll-target="#gwas-data"><span class="header-section-number">2.2.2</span> GWAS data</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Data</span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>The sclerostin pQTLs and GWAS results for are processed using the scripts sourced below. The linkage disequilibrium (LD) matrix of genetic variants in the <em>SOST</em> region (100KB downstream and 125KB upstream<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a> of <em>SOST</em>) is computed using data from 1000 Genomes <span class="citation" data-cites="genomes2015">(<a href="09_references.html#ref-genomes2015" role="doc-biblioref">1000 Genomes Project Consortium et al. 2015</a>)</span>.</p>
<section id="datasets" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="datasets"><span class="header-section-number">2.1</span> Datasets</h2>
<section id="sclerostin-pqtls" class="level3" data-number="2.1.1">
<h3 data-number="2.1.1" class="anchored" data-anchor-id="sclerostin-pqtls"><span class="header-section-number">2.1.1</span> Sclerostin pQTLs</h3>
<p>The <em>cis</em> sclerostin pQTLs in <a href="#tbl-pqtls">Table&nbsp;<span>2.1</span></a> were extracted from Supplementary Table 2 in <span class="citation" data-cites="zheng2023">Zheng et al. (<a href="09_references.html#ref-zheng2023" role="doc-biblioref">2023</a>)</span>.</p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-pqtls" class="anchored">
<table class="table" data-quarto-postprocess="true" style="margin-left: auto; margin-right: auto;">
<caption>Table&nbsp;2.1: cis pQTLs</caption>
<tbody>
<tr class="header">
</tr>

</tbody>
</table>
</div>


</div>
<div class="cell-output-display">

<div class="reactable html-widget html-fill-item" id="htmlwidget-c90549a99cdfa2f586c5" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-c90549a99cdfa2f586c5">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"rsid":["rs66838809","rs1107747","rs4793023","rs80107551","rs76449013"],"chr":[17,17,17,17,17],"pos":[41798621,41774270,41800656,41808374,41791049],"ea":["A","T","A","T","T"],"oa":["G","C","C","C","C"],"eaf":[0.079,0.61,0.381,0.08,0.06],"beta":[-0.088,0.042,-0.04,-0.072,-0.082],"se":[0.015,0.008,0.008,0.014,0.017],"pvalue":[1.45e-09,9.29e-08,2.52e-07,4.06e-07,8.74e-07],"n":[29381,29381,29381,29381,29381]},"columns":[{"id":"rsid","name":"rsID","type":"character","minWidth":100,"align":"left"},{"id":"chr","name":"Chr","type":"numeric","minWidth":50,"align":"left"},{"id":"pos","name":"Position","type":"numeric","minWidth":90,"align":"left"},{"id":"ea","name":"Effect allele","type":"character","minWidth":80,"align":"left"},{"id":"oa","name":"Other allele","type":"character","minWidth":80,"align":"left"},{"id":"eaf","name":"Effect allele frequency","type":"numeric","minWidth":100,"align":"left"},{"id":"beta","name":"Effect size","type":"numeric","minWidth":80,"align":"left"},{"id":"se","name":"Standard error","type":"numeric","minWidth":80,"align":"left"},{"id":"pvalue","name":"P-value","type":"numeric","minWidth":80,"align":"left"},{"id":"n","name":"Number of samples","type":"numeric","minWidth":100,"align":"left"}],"searchable":true,"bordered":true,"striped":true,"dataKey":"deeffe2a9e21b071b2050eb6cd1c9347"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p><br style="line-height: 5px"></p>
<div class="cell">
<div class="cell-output-display">
<button class="btn btn-primary" onclick="fetch('data:application/vnd.openxmlformats-officedocument.spreadsheetml.sheet;base64,UEsDBBQAAAAIAAAAIQA4nYbYPgEAAAcEAAATAAAAW0NvbnRlbnRfVHlwZXNdLnhtbK2Ty27DIBBF9/0KxLYyJF1UVRUniz6WbRbpB1AYxygYEDNJk78vtpNIrdI8lG6MzNy55w6C0WTdOLaChDb4kg/FgDPwOhjr5yX/mL0WD5whKW+UCx5KvgHkk/HNaLaJgCw3eyx5TRQfpURdQ6NQhAg+V6qQGkX5N81lVHqh5iDvBoN7qYMn8FRQ68HHo2eo1NIRe1nn7T5IAoecPfXCllVyFaOzWlGuy5U3vyjFliByZ6fB2ka8zQIuDxLayt+Abd97PplkDbCpSvSmmqySJuhpChFl1ovjLgdihqqyGrLHssktAtpABkwRsyUksrDPfJStQ4LL4bszarvPJK6dRNo4wKtHxZhAGawBqHGiNz1BpnyfoP8Or+Z3NieAXyEtPkNY/Pew7SoaZf0Z/E6Msluun/pnkL3/Lofs3vH4G1BLAwQUAAAACAAAACEA8p9J2ukAAABLAgAACwAAAF9yZWxzLy5yZWxzrZLBTsMwDEDvfEXk+5puSAihpbsgpN0mND7AJG4btY2jxIPu74mQQAyNaQeOceznZ8vrzTyN6o1S9hwMLKsaFAXLzofOwMv+aXEPKgsGhyMHMnCkDJvmZv1MI0qpyb2PWRVIyAZ6kfigdbY9TZgrjhTKT8tpQinP1OmIdsCO9Kqu73T6yYDmhKm2zkDauiWo/THSNWxuW2/pke1hoiBnWvzKKGRMHYmBedTvnIZX5qEqUNDnXVbXu/w9p55I0KGgtpxoEVOpTuLLWr91HNtdCefPjEtCt/+5HJqFgiN3WQlj/DLSJzfQfABQSwMEFAAAAAgAAAAhAOVEG6PVAAAALAIAABoAAAB4bC9fcmVscy93b3JrYm9vay54bWwucmVsc62Rz4oCMQyH7z5FyX0nMwoiYsfLsuBV9AFKJ/MHZ9rSZHedt7couiuIePAUfgn58kFW6+PQqx+K3HmnochyUOSsrzrXaNjvvj4WoFiMq0zvHWkYiWFdTlZb6o2kHW67wCpBHGtoRcISkW1Lg+HMB3JpUvs4GEkxNhiMPZiGcJrnc4z/GVDeMdWm0hA3VQFqNwZ6he3rurP06e33QE4enMBfHw/cEkmCmtiQaLi1GM+lyBIV8LHM9J0yknbpT+QcL82nDrN3OrCMfXroTeKSr+fx7snlCVBLAwQUAAAACAAAACEAba+ou+wCAADqCgAAGAAAAHhsL3dvcmtzaGVldHMvc2hlZXQxLnhtbI2W0W7aMBSG7/cUUe6bOIljJ1VI1Y2VgjRpWrvdm+BA1BAj29Dt7ecQcMGzw7hAsX//5/h8ORgXD7+3rXegXDSsm/hRAHyPdhVbNd164v98fbrLfE9I0q1Iyzo68f9Q4T+Un4p3xt/EhlLpqQCdmPgbKXf3YSiqDd0SEbAd7ZRSM74lUg35OhQ7TsnqaNq2YQwACrek6fwhwj3/nxisrpuKTlm139JODkE4bYlU2xebZif8slg1Suvr8TitJ/5jdL9AflgWx8y/GvouLp49SZYvtKWVpCtVv+/1hS0Ze+vFuZoCvTX8x/t03NR37q1oTfat/MHen2mz3kgVJNXZpkSSsuDs3ePH4OL4Xe2FZNshQj9RFlWvP+oFarbp2qajL5IrtVFJZale0aoIpdpMPw6rk+vzuKvacIvpy7hpx4TFNB03UWLxfB33MJvn6Vae2mKajZuWVNpSPY+7BLV45jfYHUi7t/kW477u2hKqptGdE+sWiZ3NgVCWZBnIbS3S+w9lhIvwcNkEwzSMcJ6hOLoWp65Mj7a37Fo8s73eIS0IAM6vc84G5U5JGc6vPtcLn3WIKL1W5qdSA5jSO2DYFoMY50n2Ue0V6ESDTpygowhgDLGNc2LnnJw5YxhjYHB2JXq1cXYt/mLjnJwgIePVzs4CgLHBVSsgM7gOSh7EueJqiIvkFleouUInV4jzBMSJjSu0c4Unrpn6F0iRwdWVyNq/rsVWrvAE6bLaASzU/QsNsNAJdlDiII0VWKPCBbwFNtVgUyfYDKiOTdPIRja1k0012SzBRi1TVyZrx7oWW8mmZ0oGpFmqweJo9GDQESJj1/NTSQFAFs7pLc5Ic0ZOzhhBmIPI2sHIzhnpEzgC0Khl6spk5exabOWMzpSMH80MfRzA5tGgPWYR80HJAgwtZJGTbHhxQ9qRNf1G+LrphNfSWlUCAux7fLhQHZ8l2x2fVEMtmVT3p/Nooy6VlPcjdTrWjMnzoL+36Wtq+RdQSwMEFAAAAAgAAAAhAIMYaiVIAQAAJgIAAA8AAAB4bC93b3JrYm9vay54bWyNUctOwzAQvPMV1t5pHmojWjWpxEtUQoBEac8m3jRWHTuyHdL+PetUKXDjtDPj3dHOerk6Nop9oXXS6BySSQwMdWmE1PscPjaP1zfAnOdacGU05nBCB6viatkbe/g05sBoXrscau/bRRS5ssaGu4lpUdNLZWzDPVG7j1xrkQtXI/pGRWkcZ1HDpYazw8L+x8NUlSzx3pRdg9qfTSwq7ml7V8vWQbGspMLtORDjbfvCG1r7qIAp7vyDkB5FDlOipsc/gu3a206qQGbxDKLiEvLNMoEV75Tf0GqjO50rnaZpFjpD11Zi736GAmXHndTC9DmkU7rsaWTJDFg/4J0UviYhi+cX7QnlvvY5zLMsDubRL/fhfmNlegj3HnBC/xTqmvYnbBeSgF2LZHAYx0quSkoTytCYTmfJHFjVKXVH2qt+NnwwCENjkuIbUEsDBBQAAAAIAAAAIQA/2O8hsQUAAFMbAAATAAAAeGwvdGhlbWUvdGhlbWUxLnhtbO1ZTY/TRhi+8ytGvoPjxA7ZFVm0ySbQwsJqN1BxnNgTe8jYY81MdsmtgmOlSlVp1Uul3nqo2iKB1Av9NdtStVTiL/T1R5LxZrJkYasWQQ6JZ/y83x9+x7ly9UHM0CERkvKkbTmXahYiic8DmoRt686gf7FlIalwEmDGE9K2pkRaV7cuXMGbKiIxQUCeyE3ctiKl0k3blj5sY3mJpySBeyMuYqxgKUI7EPgI2MbMrtdqTTvGNLFQgmPgens0oj5Bg4yltTVj3mPwlSiZbfhMHPi5RJ0ixwZjJ/uRU9llAh1i1rZATsCPBuSBshDDUsGNtlXLP5a9dcWeEzG1glaj6+efkq4kCMb1nE6Ewzmh03c3Lu/M+dcL/su4Xq/X7TlzfjkA+z5Y6ixh3X7L6cx4aqDicpl3t+bV3Cpe499Ywm90Oh1vo4JvLPDuEr5Va7rb9QreXeC9Zf07291us4L3FvjmEr5/eaPpVvE5KGI0GS+hs3jOIzOHjDi7boS3AN6aJcACZWvZVdAnalWuxfg+F30A5MHFiiZITVMywj7gujgeCoozAXiTYO1OseXLpa1MFpK+oKlqWx+nGCpiAXn1/MdXz5+iV8+fHD98dvzwl+NHj44f/mwgvI6TUCd8+f0Xf3/7Kfrr6XcvH39lxksd//tPn/3265dmoNKBL75+8sezJy+++fzPHx4b4NsCD3X4gMZEolvkCO3zGGwzCCBDcTaKQYRphQJHgDQAeyqqAG9NMTPhOqTqvLsCGoAJeG1yv6LrQSQmihqAN6K4AtzlnHW4MJpzI5OlmzNJQrNwMdFx+xgfmmR3T4S2N0khk6mJZTciFTX3GEQbhyQhCmX3+JgQA9k9Sit+3aW+4JKPFLpHUQdTo0sGdKjMRNdpDHGZmhSEUFd8s3sXdTgzsd8hh1UkFARmJpaEVdx4DU8Ujo0a45jpyJtYRSYlD6bCrzhcKoh0SBhHvYBIaaK5LaYVdW9g6ETGsO+yaVxFCkXHJuRNzLmO3OHjboTj1KgzTSId+5EcQ4pitMeVUQlerZBsDXHAycpw36VEna2s79AwMidIdmciyq5d6b8xTU5rxoxCN/7QjGfwbXg0mUriZAtehXsHG+8OniR7BHL9Q9/90Hffx767qpbX7baLBmvrc3HOL145JI8oYwdqyshNmbdmCUoHfdjMFznRfCZPI7gsxVVwocD5NRJcfUJVdBDhFMQ4uYRQlqxDiVIu4SRgreSdHycpGJ/vebMzIKCx2uVBsd3Qz4ZzNvkqlLqgRsZgXWGNy28nzCmAa0pzPLM071RptuZNqAaEs4O/06wXoiFjMCNB5veCwSws5x4iGeGAlDFyjIY4jTXd1nq91zRpG423k7ZOkHRx7gpx3jlEqbYUJXu5HFlSXaEj0Mqrexbycdq2RjBJwWWcAj+ZNSDMwqRt+ao05bXFfNJgc1o6tZUGV0SkQqodLKOCKr81e3WSLPSve27mh/MxwNCN1tOi0XL+Qy3sk6EloxHx1YqdxbK8xyeKiIMoOEJDNhH7GPR2i+wKqIRnRn22EFChbpl41covq+DkK5qyOjBLI1z2pJYW+wKeX891yFeaevYK3d/QlMY5muK9v6ZkmQtjayPID1QwBgiMshxtW1yoiEMXSiPq9wUMDrks0AtBWWQqIZa9b850JYeLvlXwKJpcGKl9GiJBodOpSBCyp0o7X8PMqevP1xmjss/M1ZVp8Tskh4QNsuptZvZbKJp1k9IROe5k0GxTdQ3D/v948nFXTD6njwcLQe5ZZhFXa/rao2Dj7VQ446O2bra47q39qE3h8IGyL2jcVPhsMd8O+D5EH80nSgSJeLFVlt98cwg6tzTjMlb/7hi1CEFrRbzPc/jUnN1Y4ezTxb25sz2Dr73TXW0vl6itHWTy1dIfT3x4H2TvwEFpwpQs3iY9gKNmd/aXAfCxF6RbF/4BUEsDBBQAAAAIAAAAIQCvzhJdwwEAAFIEAAANAAAAeGwvc3R5bGVzLnhtbL1U34ucMBB+718R8t5zFXrcFfUohYVCWw5uC32NGjUwSSQZF72//iZG3V1oOehDX8zMNzPf/HCS/GnSwM7SeWVNwdO7A2fS1LZRpiv4r9Px4wNnHoVpBFgjCz5Lz5/KD7nHGeRLLyUyYjC+4D3i8DlJfN1LLfydHaQhS2udFkiq6xI/OCkaH4I0JNnhcJ9ooQwv89Ya9Ky2o8GCZytQ5v6VnQVQWSlPyry2YB1Doqc6FsQILaPHVwGqciqArdAK5ghnAVgqWv20MtYFMIkZ4rdK/keu5fAUpABumyWgzAeBKJ05ksJW+TQPlN7Q4CPN4veOd+fEnGafrgKWg/JW1jX0o7fMKd+gMgfZIgU41fXhRDskwYhoNQmNEp01AgLlFrEKRFtLgJewDb/bG+6pZWbUR43fmoLTWoXuN5EKWsVIE5XAf80Wua9os3+iZVO78/8tOn0/molhgPloY39R+wKqM1puLYtNZb116pVcwx7UBMi4CFO7trh3t/R6M7cdZWHlCv4z3CC4qqMaFaAyf5gZcTbTZVyLFUVFF/UmC3E0shUj4Gk3Fvwi/5CNGvXj7vWszhZXr4v8PSxLer9UcHkNyjdQSwMEFAAAAAgAAAAhANoMtWMlAQAAUAIAABEAAABkb2NQcm9wcy9jb3JlLnhtbJ2SzWrDMBCE730Ko7sty05LELYDbcmpgUJTWnoT0iYRtX6Q1Dp++yqO4ySQU4+rmf12dlG12Ks2+QXnpdE1IlmOEtDcCKm3NXpfL9M5SnxgWrDWaKhRDx4tmruKW8qNg1dnLLggwScRpD3ltka7ECzF2PMdKOaz6NBR3BinWIil22LL+DfbAi7y/AErCEywwPABmNqJiEak4BPS/rh2AAiOoQUFOnhMMoLP3gBO+ZsNg3LhVDL0Fm5aT+Lk3ns5Gbuuy7pysMb8BH+uXt6GVVOpD6figJpKcModsGBcU+HLIh6uZT6s4ok3EsRjH/Ubb+Mixz4QSQxAj3FPykf59LxeoqbIi1makzQv1mROyT2dlV+HkVf9Z6Aah/ybeAIcc19/guYPUEsDBBQAAAAIAAAAIQBeuqfTdwEAABADAAAQAAAAZG9jUHJvcHMvYXBwLnhtbJ2SwU7rMBBF93xF5D11UiH0VDlGqIBY8ESlFlgbZ9JYOLblGaKWr8dJ1ZACK7K6M3N1fTK2uNq1NusgovGuZMUsZxk47SvjtiV72tyd/2MZknKVst5ByfaA7EqeiVX0ASIZwCwlOCxZQxQWnKNuoFU4S2OXJrWPraJUxi33dW003Hj93oIjPs/zSw47AldBdR7GQHZIXHT019DK654Pnzf7kPKkuA7BGq0o/aT8b3T06GvKbncarODToUhBa9Dv0dBe5oJPS7HWysIyBctaWQTBvxriHlS/s5UyEaXoaNGBJh8zNB9pa3OWvSqEHqdknYpGOWIH26EYtA1IUb74+IYNAKHgY3OQU+9UmwtZDIYkTo18BEn6FHFjyAI+1isV6RfiYko8MLAJ47rnK37wHU/6lr30bVAuLZCP6sG4N3wKG3+jCI7rPG2KdaMiVOkGxnWPDXGfuKLt/ctGuS1UR8/PQX/5z4cHLov5LE/fcOfHnuBfb1l+AlBLAQIAABQAAAAIAAAAIQA4nYbYPgEAAAcEAAATAAAAAAAAAAEAAAAAAAAAAABbQ29udGVudF9UeXBlc10ueG1sUEsBAgAAFAAAAAgAAAAhAPKfSdrpAAAASwIAAAsAAAAAAAAAAQAAAAAAbwEAAF9yZWxzLy5yZWxzUEsBAgAAFAAAAAgAAAAhAOVEG6PVAAAALAIAABoAAAAAAAAAAQAAAAAAgQIAAHhsL19yZWxzL3dvcmtib29rLnhtbC5yZWxzUEsBAgAAFAAAAAgAAAAhAG2vqLvsAgAA6goAABgAAAAAAAAAAQAAAAAAjgMAAHhsL3dvcmtzaGVldHMvc2hlZXQxLnhtbFBLAQIAABQAAAAIAAAAIQCDGGolSAEAACYCAAAPAAAAAAAAAAEAAAAAALAGAAB4bC93b3JrYm9vay54bWxQSwECAAAUAAAACAAAACEAP9jvIbEFAABTGwAAEwAAAAAAAAABAAAAAAAlCAAAeGwvdGhlbWUvdGhlbWUxLnhtbFBLAQIAABQAAAAIAAAAIQCvzhJdwwEAAFIEAAANAAAAAAAAAAEAAAAAAAcOAAB4bC9zdHlsZXMueG1sUEsBAgAAFAAAAAgAAAAhANoMtWMlAQAAUAIAABEAAAAAAAAAAQAAAAAA9Q8AAGRvY1Byb3BzL2NvcmUueG1sUEsBAgAAFAAAAAgAAAAhAF66p9N3AQAAEAMAABAAAAAAAAAAAQAAAAAASREAAGRvY1Byb3BzL2FwcC54bWxQSwUGAAAAAAkACQA+AgAA7hIAAAAA').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      a.style.display = 'none'; // Hide the link
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'pqtls.xlsx';
      a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
      return false;
    });" has_icon="TRUE"><i class="fa fa-save"></i> Download as xlsx</button>
</div>
</div>
<details>
<summary>
README
</summary>
<ul>
<li><code>rsid</code> - rsID<br>
</li>
<li><code>chr</code> - chromosome<br>
</li>
<li><code>pos</code> - position (build 37)<br>
</li>
<li><code>ea</code> - effect allele<br>
</li>
<li><code>oa</code> - other allele<br>
</li>
<li><code>eaf</code> - effect allele frequency<br>
</li>
<li><code>beta</code> - effect size</li>
<li><code>se</code> - standard error of effect size</li>
<li><code>pvalue</code> - <em>p</em>-pvalue</li>
<li><code>n</code> - number of samples</li>
</ul>
</details>
</section>
<section id="study-information" class="level3" data-number="2.1.2">
<h3 data-number="2.1.2" class="anchored" data-anchor-id="study-information"><span class="header-section-number">2.1.2</span> Study information</h3>
<p>The GWAS studies used in the analyses<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a> are presented in <a href="#tbl-gwas-study-info">Table&nbsp;<span>2.2</span></a>.<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a></p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-gwas-study-info" class="anchored">
<table class="table" data-quarto-postprocess="true" style="margin-left: auto; margin-right: auto;">
<caption>Table&nbsp;2.2: GWAS study information</caption>
<tbody>
<tr class="header">
</tr>

</tbody>
</table>
</div>


</div>
<div class="cell-output-display">

<div class="reactable html-widget html-fill-item" id="htmlwidget-a35837d210b130a1d442" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-a35837d210b130a1d442">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"id":["<a href='https://www.ebi.ac.uk/gwas/studies/GCST011365/'>GCST011365<\/a>","<a href='https://www.ebi.ac.uk/gwas/studies/GCST005194/'>GCST005194<\/a>","<a href='https://www.ebi.ac.uk/gwas/studies/GCST90132315/'>GCST90132315<\/a>","<a href='https://www.ebi.ac.uk/gwas/studies/GCST005843/'>GCST005843<\/a>","<a href='https://www.ebi.ac.uk/gwas/studies/GCST90104535/'>GCST90104535<\/a>","<a href='https://www.ebi.ac.uk/gwas/studies/GCST005842/'>GCST005842<\/a>","<a href='https://www.ebi.ac.uk/gwas/studies/GCST90104536/'>GCST90104536<\/a>","<a href='https://www.ebi.ac.uk/gwas/studies/GCST90104538/'>GCST90104538<\/a>","<a href='https://www.ebi.ac.uk/gwas/studies/GCST90104537/'>GCST90104537<\/a>","<a href='https://www.ebi.ac.uk/gwas/studies/GCST006867/'>GCST006867<\/a>","<a href='https://diagram-consortium.org/downloads.html'>GCST90132183<\/a>","<a href='https://pan.ukbb.broadinstitute.org/downloads'>UKB200021065<\/a>","<a href='https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000930.v10.p1'>PHS000930AAC<\/a>","<a href='http://www.nealelab.is/uk-biobank'>UKB30780<\/a>","<a href='http://www.nealelab.is/uk-biobank'>UKB30760<\/a>","<a href='http://www.nealelab.is/uk-biobank'>UKB30870<\/a>","<a href='http://www.nealelab.is/uk-biobank'>UKB30630<\/a>","<a href='http://www.nealelab.is/uk-biobank'>UKB30640<\/a>","<a href='https://www.ebi.ac.uk/gwas/studies/GCST006979/'>GCST006979<\/a>","<a href='https://www.ebi.ac.uk/gwas/studies/GCST006980/'>GCST006980<\/a>"],"pmid":["<a href='https://pubmed.ncbi.nlm.nih.gov/33532862'>33532862<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/29212778'>29212778<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/36474045'>36474045<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/29531354'>29531354<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/36180795'>36180795<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/29531354'>29531354<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/36180795'>36180795<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/36180795'>36180795<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/36180795'>36180795<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/30054458'>30054458<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/35551307'>35551307<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/25826379'>25826379<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/31659325'>31659325<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/25826379'>25826379<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/25826379'>25826379<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/25826379'>25826379<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/25826379'>25826379<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/25826379'>25826379<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/30598549'>30598549<\/a>","<a href='https://pubmed.ncbi.nlm.nih.gov/30598549'>30598549<\/a>"],"trait":["Myocardial infarction","Coronary artery disease","Coronary artery disease","Ischemic stroke","Ischemic stroke","Cardioembolic stroke","Cardioembolic stroke","Large artery stroke","Small vessel stroke","Type 2 diabetes","Type 2 diabetes","Hypertension","Abdominal aortic calcification","LDL cholesterol","HDL cholesterol","Triglycerides","Apolipoprotein A-I","Apolipoprotein B","Heel bone mineral density","Fracture"],"abbr":["MI","CAD","CAD","IS","IS","CES","CES","LAS","SVS","T2D","T2D","HTN","AAC","LDL","HDL","TG","APOA","APOB","BMD","FRAC"],"ancestry":["Trans-ethnic","Trans-ethnic","Trans-ethnic","Trans-ethnic","Trans-ethnic","Trans-ethnic","Trans-ethnic","Trans-ethnic","Trans-ethnic","European","Trans-ethnic","Trans-ethnic","European","European","European","European","European","European","European","European"],"n":[639221,547261,1378170,514791,1590566,463456,1516688,1513117,1517518,659316,1339889,432029,9417,343621,315133,343992,313387,342590,426824,426795],"n_cases":[61505,122733,210842,60341,86668,9006,12790,9219,13620,62892,180834,112434,"NA","NA","NA","NA","NA","NA","NA",53184],"n_controls":[577716,424528,1167328,454450,1503898,454450,1503898,1503898,1503898,596424,1159055,319595,"NA","NA","NA","NA","NA","NA","NA",373611]},"columns":[{"id":"id","name":"Dataset ID","type":"character","html":true,"minWidth":100,"align":"left"},{"id":"pmid","name":"PubMed ID","type":"character","html":true,"minWidth":80,"align":"left"},{"id":"trait","name":"Phenotype","type":"character","minWidth":140,"align":"left"},{"id":"abbr","name":"Abbreviation","type":"character","minWidth":100,"align":"left"},{"id":"ancestry","name":"Ancestry","type":"character","minWidth":100,"align":"left"},{"id":"n","name":"Number of samples","type":"numeric","minWidth":80,"align":"left"},{"id":"n_cases","name":"Number of cases","type":"numeric","minWidth":80,"align":"left"},{"id":"n_controls","name":"Number of controls","type":"numeric","minWidth":80,"align":"left"}],"searchable":true,"bordered":true,"striped":true,"dataKey":"0e404efbf5ec3e30b21d7cf24f3c9361"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<button class="btn btn-primary" onclick="fetch('data:application/vnd.openxmlformats-officedocument.spreadsheetml.sheet;base64,UEsDBBQAAAAIAAAAIQA4nYbYPgEAAAcEAAATAAAAW0NvbnRlbnRfVHlwZXNdLnhtbK2Ty27DIBBF9/0KxLYyJF1UVRUniz6WbRbpB1AYxygYEDNJk78vtpNIrdI8lG6MzNy55w6C0WTdOLaChDb4kg/FgDPwOhjr5yX/mL0WD5whKW+UCx5KvgHkk/HNaLaJgCw3eyx5TRQfpURdQ6NQhAg+V6qQGkX5N81lVHqh5iDvBoN7qYMn8FRQ68HHo2eo1NIRe1nn7T5IAoecPfXCllVyFaOzWlGuy5U3vyjFliByZ6fB2ka8zQIuDxLayt+Abd97PplkDbCpSvSmmqySJuhpChFl1ovjLgdihqqyGrLHssktAtpABkwRsyUksrDPfJStQ4LL4bszarvPJK6dRNo4wKtHxZhAGawBqHGiNz1BpnyfoP8Or+Z3NieAXyEtPkNY/Pew7SoaZf0Z/E6Msluun/pnkL3/Lofs3vH4G1BLAwQUAAAACAAAACEA8p9J2ukAAABLAgAACwAAAF9yZWxzLy5yZWxzrZLBTsMwDEDvfEXk+5puSAihpbsgpN0mND7AJG4btY2jxIPu74mQQAyNaQeOceznZ8vrzTyN6o1S9hwMLKsaFAXLzofOwMv+aXEPKgsGhyMHMnCkDJvmZv1MI0qpyb2PWRVIyAZ6kfigdbY9TZgrjhTKT8tpQinP1OmIdsCO9Kqu73T6yYDmhKm2zkDauiWo/THSNWxuW2/pke1hoiBnWvzKKGRMHYmBedTvnIZX5qEqUNDnXVbXu/w9p55I0KGgtpxoEVOpTuLLWr91HNtdCefPjEtCt/+5HJqFgiN3WQlj/DLSJzfQfABQSwMEFAAAAAgAAAAhAOVEG6PVAAAALAIAABoAAAB4bC9fcmVscy93b3JrYm9vay54bWwucmVsc62Rz4oCMQyH7z5FyX0nMwoiYsfLsuBV9AFKJ/MHZ9rSZHedt7couiuIePAUfgn58kFW6+PQqx+K3HmnochyUOSsrzrXaNjvvj4WoFiMq0zvHWkYiWFdTlZb6o2kHW67wCpBHGtoRcISkW1Lg+HMB3JpUvs4GEkxNhiMPZiGcJrnc4z/GVDeMdWm0hA3VQFqNwZ6he3rurP06e33QE4enMBfHw/cEkmCmtiQaLi1GM+lyBIV8LHM9J0yknbpT+QcL82nDrN3OrCMfXroTeKSr+fx7snlCVBLAwQUAAAACAAAACEAVm17PI4KAAB2OwAAGAAAAHhsL3dvcmtzaGVldHMvc2hlZXQxLnhtbL2bW1PjuBaF38+vcOXdseW7u4CpcEu4NRQwU+c8UYqtJB5sKyUr0MyvH/mSBMzeCt0N56U7Rl47kvV5RdZK9v74UeTGExNVxsv9ARnaA4OVCU+zcr4/+PP+1IwGRiVpmdKcl2x/8MKqwR8H/9l75uKxWjAmDVWgrPYHCymX3yyrShasoNWQL1mpWmZcFFSqQzG3qqVgNG1ERW45th1YBc3KQVvhm/hIDT6bZQk75smqYKVsiwiWU6m6Xy2yZTU42Esz1VaPxxBstj8YkW9XDhlYB3vNW/+Vsefq1WtD0ukdy1kiWaouwMCoRzbl/LFuPFN/smup9U572vTqRhgpm9FVLm/584Rl84VURfzNux1TSQ/2BH82RFO8av5NVpXkRVuh/sPBXlK3jzYnqL9mZZ6V7E4K1ZqpN5UHWbpnSdWV+shKOs2hXlPxlUgYoDvS65YF+G7HehVdyQUXgO5Er1PHj4DqVK+SgmYSkI13dHI6hbo42aEqE1ZJ8QIoz/TKEpCc75A8JLRiFSC82CnkpRQ8h7SXeu2qBK/mlV41y+n8rcpStG+QdzZsO0iB8dHdvU2IG/gQ3ToVRHX9fk8Hruu7ThQ4e9bTa3ixYhMqZEZzapyPIHYxWedWM7kcsmk2pMlw9WgtV1MrVTf9tJ5Aa/5MK6taFQUVLw/KRmVWySyprG7Qtk3M9qUyMtvaXgprQUXBy6xiqbUejLltNk9Orx+Uwg6IYx6usjx1w+FsKKun4fwf6EbChnD1whMqUjV4IytnVCS1j0K3FFrgDLqVsLPvBS0rk8lFmSXQjdTOXuDGjkPezt1510R823/bctG2+GEYkuBt0yXWj5zPjetbCHZM8T8N4+6GcVfHuO2T2IMY16kgxt1mwE7sECcMox7jWLEnWqpPKmEo1itp3ECcY9Lf5dz2N5zbwYbz5nK85nw9IHPbvOHcVX+efoBzbAhHXPBS9c1QdzpT/6XqHVXHIdLREqNjCHXs9F2ot5Poe6ET9FFvm4jjhK7bY71t8hzPd3oTf4l1BGcdU+hY9zasexpqY5u4jktAR9fpINq91tEDL/Rsr3fvH2PFRoLOaWFcjCHOMdFvct4Mek16feCuWV9fjjcHDw3P49ujhWJaEQ3hjPX0J3BGS8A4Y6fvwrmdJaLuUxLaPZ7bNofYkdf7RL7oZCQI3XdAY13BgcYU3zVA+xugfb15R54L4Yyprk7Go7v72+uLEwhqv7Nw3yWu7/WgRkvSPHs0oJGfYJKvsm51Md5adzsQc9tsTm5q53aIZ3/IubERnDUPg1miHkIFfwQRR6V3EOHY2bsIb6fMJ14Y9w27bQps1+u1XLQtnu95fu+uuMT6geONKXR+HWzwDvR+rbzVBf1ap4PQDjq/JpEdxn2/xopdZdVCUANcfmOa37dr23tl1zXqtvX6arw56Ns1gjHW2Q9gjEpBjLGzd2HcTg/xY9sPesvl87YtCoKg58UXa5XtRnHfp7Ge4CBjCp1PhxuQw10+7UAYYyq9T4c6n0ZL4j6NSb7Opx29TzvNErt+EI3D4ENGjQ3hqH6O5KyY8lyLOao/ATnHTt/FeTtzXuB6fh/ztilWV6pHeYi7NdYNHHJMoXPraAN5tNutAwhznQ4CPNK5NVZM59aY5qvdOnhz8EG3xjr7UZZRPcwydvoulqPOfYky5p77nndtThj3kL2INJ6N9QTHGVPoPDve4BzvxjmCcNbpIJxjHc5YMR3OmOarcY7eHHwQZ6yzl1TM2fpZEacZlY9AmrHTd9Ecr2l2CQl7NLdtsUPiHsyxBmasIzjMmEIHM7G3MY29G+cQzGp0QjCksXVAo+V0RKOir0Y6fHPwQaTR3t4VNK+Dy6piuYZpXP8XCDV6/i6qu5lSWIc+6Zv0utENnL5Lb3QQ2WhvcLRRiZbtVxEkFvR0i8sogMnWyUCySUu2WoWq5VZ/Fxst998VQ7DGFL+9tA62S+twu7SuL8SblKYbiLltXu9eq2n/2B4IOob7lyUzHCPN6JRJMBcc41oH3OhDzz9ZCb5kFEqBzrpJC/zY7Wct5+s2J4r7u3xdkx8HnuP1EUczYRxxTKJbWpNtEkl0oWKzPUsicK8PFR6fjca3oyuQ8y6R9H31gWeHfc7RXI0u6N+0RFjXJZKVgl2BMhe0MBNeVlzIbFU0X55I+XOZc5pWw4UscpBANLn7AIGoFiHwV1PC7pIS142jKO5j2DVGduR6fQ67tmb3w++D+PNhISrReu02LiRYAvPnxaHTbJ7acCqOCm+UEqSwywz9yAncMO5TiJajpakqQiVPUNEawqUSrx6n5krNJa2kSYaVO6QF/YeX9LkaJryoPbe22+phllP5MMtyZcYJlWzORZbQ3GwugjnlcvFQsR+sMusL0hjoFHFQrFMTxa9a/zZfCwLhRYX330F4fzX362bCcx3becdul/wRx3vPbtvmktjvrwMv0c5o0P2V8I9s0z+CZS03kzs1abFrj0ZHILpoIDUZ3Y7BTTjSRYCk/txx3i2CsYJXNF9wqZbB4E4cKlvj+/z8PCwTtWIo82JYZovhnD9ZS8H/ZolUSwa6tJJ5Zk6z0qrkKn0ZqqM/mlcPWbq/XFTtVRg+EXu4JCCraHg5TXmRlTQ3aG3eiaFuhSSbqTsC+3YGXgqchAl6vvbDv52G2Os/tV2i5e4g479CT9eytw3qCBaEKKtSH7GRDXKnEYHQ+VrHxKp9ZzRnBmyYupCuJm4q1GdzY5lrX9R5p+CrMnUsmqaZzJ6YqVyxsprhP2SilMN6UTvMiuVKsvThyR1uXXT4RAUXKRNaJ8V6e3l8aSQLrqxaMsGhVcRYpwVxxM7X4thOkOuphyrSBxKNBG+/Q08jV6hAi+Q2XCNYpNEiGcBIakQgkoEWSTRV0SCpy9Y+D8ngk5DEejv5AJIaLYgkdr4WyS77rPe23D6SaLyHIYkJtEhuYzKCBRANklEII6kRgUiGWiSxajokdRHZpyGphv85SOLxVDbPXxImshR5WkKV0HeWJujpWh7DtUXG/WfyS7QgyuOvBFpkm2gRLEFoeAxcmEeNCOQx0vKIhhgaHnWB1qfxqIb/OTxivR0teZ4tuVq/SpaVxsiEvrc7xuU319BOxAQVaLHsQkdlktG7pSRWEMUSE2ix3CZTBMsCWiw9GEuNCMQy1mKJxhEaLHXB1Odh6X0Sllhve1hCQx3j4ptrSDBBBVoou+jQc/x+knqJFkShxAQ6KJ1twOTociLbDuKaofdY4rKT02so7jhyunzJ9uPI9/pgovWuuBBZhf1c4osCJnQnXl0N6zDjU1o+OuahUARXC/OwSM1RZR6ZY1XSvFsVd/U9MJQ/JPYTCazbE8ZyY8pLZqiHcibUY3la7yNJ6KdAY7TK4RW4CYqer2O1mzXPCaJ3++loQYxVVKBldRsYOTsCoxh+HsdlKKtEyypWT8/q/zs1UlfjPaungiajn6QV63hdS64EGIbiolt4nwgVaOkkazrfJdbnXZvvkqi/wbme3tANSP95Hu2H5oc8PxUSWa9+rLmkc3ZFxTwrKyNnM1XEHqq1r2h/29m8lnzZvPIHyhik5MX6aMGo+iisj9yBMePqU607qH9CuvnJ7MG/UEsDBBQAAAAIAAAAIQCDGGolSAEAACYCAAAPAAAAeGwvd29ya2Jvb2sueG1sjVHLTsMwELzzFdbeaR5qI1o1qcRLVEKARGnPJt40Vh07sh3S/j3rVClw47Qz493Rznq5OjaKfaF10ugckkkMDHVphNT7HD42j9c3wJznWnBlNOZwQger4mrZG3v4NObAaF67HGrv20UUubLGhruJaVHTS2Vswz1Ru49ca5ELVyP6RkVpHGdRw6WGs8PC/sfDVJUs8d6UXYPan00sKu5pe1fL1kGxrKTC7TkQ4237whta+6iAKe78g5AeRQ5ToqbHP4Lt2ttOqkBm8Qyi4hLyzTKBFe+U39BqozudK52maRY6Q9dWYu9+hgJlx53UwvQ5pFO67GlkyQxYP+CdFL4mIYvnF+0J5b72OcyzLA7m0S/34X5jZXoI9x5wQv8U6pr2J2wXkoBdi2RwGMdKrkpKE8rQmE5nyRxY1Sl1R9qrfjZ8MAhDY5LiG1BLAwQUAAAACAAAACEAP9jvIbEFAABTGwAAEwAAAHhsL3RoZW1lL3RoZW1lMS54bWztWU2P00YYvvMrRr6D48QO2RVZtMkm0MLCajdQcZzYE3vI2GPNTHbJrYJjpUpVadVLpd56qNoigdQL/TXbUrVU4i/09UeS8WayZGGrFkEOiWf8vN8ffse5cvVBzNAhEZLypG05l2oWIonPA5qEbevOoH+xZSGpcBJgxhPStqZEWle3LlzBmyoiMUFAnshN3LYipdJN25Y+bGN5iackgXsjLmKsYClCOxD4CNjGzK7Xak07xjSxUIJj4Hp7NKI+QYOMpbU1Y95j8JUomW34TBz4uUSdIscGYyf7kVPZZQIdYta2QE7AjwbkgbIQw1LBjbZVyz+WvXXFnhMxtYJWo+vnn5KuJAjG9ZxOhMM5odN3Ny7vzPnXC/7LuF6v1+05c345APs+WOosYd1+y+nMeGqg4nKZd7fm1dwqXuPfWMJvdDodb6OCbyzw7hK+VWu62/UK3l3gvWX9O9vdbrOC9xb45hK+f3mj6VbxOShiNBkvobN4ziMzh4w4u26EtwDemiXAAmVr2VXQJ2pVrsX4Phd9AOTBxYomSE1TMsI+4Lo4HgqKMwF4k2DtTrHly6WtTBaSvqCpalsfpxgqYgF59fzHV8+folfPnxw/fHb88JfjR4+OH/5sILyOk1AnfPn9F39/+yn66+l3Lx9/ZcZLHf/7T5/99uuXZqDSgS++fvLHsycvvvn8zx8eG+DbAg91+IDGRKJb5Ajt8xhsMwggQ3E2ikGEaYUCR4A0AHsqqgBvTTEz4Tqk6ry7AhqACXhtcr+i60EkJooagDeiuALc5Zx1uDCacyOTpZszSUKzcDHRcfsYH5pkd0+EtjdJIZOpiWU3IhU19xhEG4ckIQpl9/iYEAPZPUorft2lvuCSjxS6R1EHU6NLBnSozETXaQxxmZoUhFBXfLN7F3U4M7HfIYdVJBQEZiaWhFXceA1PFI6NGuOY6cibWEUmJQ+mwq84XCqIdEgYR72ASGmiuS2mFXVvYOhExrDvsmlcRQpFxybkTcy5jtzh426E49SoM00iHfuRHEOKYrTHlVEJXq2QbA1xwMnKcN+lRJ2trO/QMDInSHZnIsquXem/MU1Oa8aMQjf+0Ixn8G14NJlK4mQLXoV7BxvvDp4kewRy/UPf/dB338e+u6qW1+22iwZr63Nxzi9eOSSPKGMHasrITZm3ZglKB33YzBc50XwmTyO4LMVVcKHA+TUSXH1CVXQQ4RTEOLmEUJasQ4lSLuEkYK3knR8nKRif73mzMyCgsdrlQbHd0M+Gczb5KpS6oEbGYF1hjctvJ8wpgGtKczyzNO9UabbmTagGhLODv9OsF6IhYzAjQeb3gsEsLOceIhnhgJQxcoyGOI013dZ6vdc0aRuNt5O2TpB0ce4Kcd45RKm2FCV7uRxZUl2hI9DKq3sW8nHatkYwScFlnAI/mTUgzMKkbfmqNOW1xXzSYHNaOrWVBldEpEKqHSyjgiq/NXt1kiz0r3tu5ofzMcDQjdbTotFy/kMt7JOhJaMR8dWKncWyvMcnioiDKDhCQzYR+xj0dovsCqiEZ0Z9thBQoW6ZeNXKL6vg5CuasjowSyNc9qSWFvsCnl/PdchXmnr2Ct3f0JTGOZrivb+mZJkLY2sjyA9UMAYIjLIcbVtcqIhDF0oj6vcFDA65LNALQVlkKiGWvW/OdCWHi75V8CiaXBipfRoiQaHTqUgQsqdKO1/DzKnrz9cZo7LPzNWVafE7JIeEDbLqbWb2WyiadZPSETnuZNBsU3UNw/7/ePJxV0w+p48HC0HuWWYRV2v62qNg4+1UOOOjtm62uO6t/ahN4fCBsi9o3FT4bDHfDvg+RB/NJ0oEiXixVZbffHMIOrc04zJW/+4YtQhBa0W8z3P41JzdWOHs08W9ubM9g6+9011tL5eorR1k8tXSH098eB9k78BBacKULN4mPYCjZnf2lwHwsRekWxf+AVBLAwQUAAAACAAAACEAr84SXcMBAABSBAAADQAAAHhsL3N0eWxlcy54bWy9VN+LnDAQfu9fEfLecxV63BX1KIWFQlsObgt9jRo1MEkkGRe9v/4mRt1daDnoQ1/MzDcz3/xwkvxp0sDO0nllTcHTuwNn0tS2UaYr+K/T8eMDZx6FaQRYIws+S8+fyg+5xxnkSy8lMmIwvuA94vA5SXzdSy38nR2kIUtrnRZIqusSPzgpGh+CNCTZ4XCfaKEML/PWGvSstqPBgmcrUOb+lZ0FUFkpT8q8tmAdQ6KnOhbECC2jx1cBqnIqgK3QCuYIZwFYKlr9tDLWBTCJGeK3Sv5HruXwFKQAbpsloMwHgSidOZLCVvk0D5Te0OAjzeL3jnfnxJxmn64CloPyVtY19KO3zCnfoDIH2SIFONX14UQ7JMGIaDUJjRKdNQIC5RaxCkRbS4CXsA2/2xvuqWVm1EeN35qC01qF7jeRClrFSBOVwH/NFrmvaLN/omVTu/P/LTp9P5qJYYD5aGN/UfsCqjNabi2LTWW9deqVXMMe1ATIuAhTu7a4d7f0ejO3HWVh5Qr+M9wguKqjGhWgMn+YGXE202VcixVFRRf1JgtxNLIVI+BpNxb8Iv+QjRr14+71rM4WV6+L/D0sS3q/VHB5Dco3UEsDBBQAAAAIAAAAIQB113olJQEAAFACAAARAAAAZG9jUHJvcHMvY29yZS54bWydks1qwzAQhO99CqO7LctxSxC2A23JqYFCU1p6E9ImEbV+kNQ6fvsqjuMkkFOPq5n9dnZRtdirNvkF56XRNSJZjhLQ3AiptzV6Xy/TOUp8YFqw1mioUQ8eLZq7ilvKjYNXZyy4IMEnEaQ95bZGuxAsxdjzHSjms+jQUdwYp1iIpdtiy/g32wIu8vwBKwhMsMDwAZjaiYhGpOAT0v64dgAIjqEFBTp4TDKCz94ATvmbDYNy4VQy9BZuWk/i5N57ORm7rsu62WCN+Qn+XL28DaumUh9OxQE1leCUO2DBuKbCl0U8XMt8WMUTbySIxz7qN97GRY59IJIYgB7jnpSP2dPzeomaIi/KNCdpXqzJnJJ7WpZfh5FX/WegGof8m3gCHHNff4LmD1BLAwQUAAAACAAAACEAXrqn03cBAAAQAwAAEAAAAGRvY1Byb3BzL2FwcC54bWydksFO6zAQRfd8ReQ9dVIh9FQ5RqiAWPBEpRZYG2fSWDi25Rmilq/HSdWQAiuyujNzdX0ytrjatTbrIKLxrmTFLGcZOO0r47Yle9rcnf9jGZJylbLeQcn2gOxKnolV9AEiGcAsJTgsWUMUFpyjbqBVOEtjlya1j62iVMYt93VtNNx4/d6CIz7P80sOOwJXQXUexkB2SFx09NfQyuueD583+5DypLgOwRqtKP2k/G909Ohrym53Gqzg06FIQWvQ79HQXuaCT0ux1srCMgXLWlkEwb8a4h5Uv7OVMhGl6GjRgSYfMzQfaWtzlr0qhB6nZJ2KRjliB9uhGLQNSFG++PiGDQCh4GNzkFPvVJsLWQyGJE6NfARJ+hRxY8gCPtYrFekX4mJKPDCwCeO65yt+8B1P+pa99G1QLi2Qj+rBuDd8Cht/owiO6zxtinWjIlTpBsZ1jw1xn7ii7f3LRrktVEfPz0F/+c+HBy6L+SxP33Dnx57gX29ZfgJQSwECAAAUAAAACAAAACEAOJ2G2D4BAAAHBAAAEwAAAAAAAAABAAAAAAAAAAAAW0NvbnRlbnRfVHlwZXNdLnhtbFBLAQIAABQAAAAIAAAAIQDyn0na6QAAAEsCAAALAAAAAAAAAAEAAAAAAG8BAABfcmVscy8ucmVsc1BLAQIAABQAAAAIAAAAIQDlRBuj1QAAACwCAAAaAAAAAAAAAAEAAAAAAIECAAB4bC9fcmVscy93b3JrYm9vay54bWwucmVsc1BLAQIAABQAAAAIAAAAIQBWbXs8jgoAAHY7AAAYAAAAAAAAAAEAAAAAAI4DAAB4bC93b3Jrc2hlZXRzL3NoZWV0MS54bWxQSwECAAAUAAAACAAAACEAgxhqJUgBAAAmAgAADwAAAAAAAAABAAAAAABSDgAAeGwvd29ya2Jvb2sueG1sUEsBAgAAFAAAAAgAAAAhAD/Y7yGxBQAAUxsAABMAAAAAAAAAAQAAAAAAxw8AAHhsL3RoZW1lL3RoZW1lMS54bWxQSwECAAAUAAAACAAAACEAr84SXcMBAABSBAAADQAAAAAAAAABAAAAAACpFQAAeGwvc3R5bGVzLnhtbFBLAQIAABQAAAAIAAAAIQB113olJQEAAFACAAARAAAAAAAAAAEAAAAAAJcXAABkb2NQcm9wcy9jb3JlLnhtbFBLAQIAABQAAAAIAAAAIQBeuqfTdwEAABADAAAQAAAAAAAAAAEAAAAAAOsYAABkb2NQcm9wcy9hcHAueG1sUEsFBgAAAAAJAAkAPgIAAJAaAAAAAA==').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      a.style.display = 'none'; // Hide the link
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'studies.xlsx';
      a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
      return false;
    });" has_icon="TRUE"><i class="fa fa-save"></i> Download as xlsx</button>
</div>
</div>
<details>
<summary>
README
</summary>
<ul>
<li><code>id</code> - dataset ID<br>
</li>
<li><code>source</code> - source of dataset<br>
</li>
<li><code>pmid</code> - PubMed ID<br>
</li>
<li><code>author</code> - author of study<br>
</li>
<li><code>link</code> - link to dataset<br>
</li>
<li><code>trait</code> - phenotype<br>
</li>
<li><code>abbr</code> - abbreviation<br>
</li>
<li><code>ancestry</code> - ancestry of study<br>
</li>
<li><code>n</code> - number of samples<br>
</li>
<li><code>n_cases</code> - number of cases<br>
</li>
<li><code>n_controls</code> - number of controls<br>
</li>
<li><code>unit</code> - unit of analysis (<code>IRNT</code> = inverse rank normal transformation)<br>
</li>
<li><code>flag</code> - flag if the dataset (or equivalent) was used by <span class="citation" data-cites="zheng2023">Zheng et al. (<a href="09_references.html#ref-zheng2023" role="doc-biblioref">2023</a>)</span> (<code>Y</code> = yes, <code>N</code> = no)</li>
</ul>
<p>Citations: <span class="citation" data-cites="xue2018">Xue et al. (<a href="09_references.html#ref-xue2018" role="doc-biblioref">2018</a>)</span>; <span class="citation" data-cites="mahajan2022">Mahajan et al. (<a href="09_references.html#ref-mahajan2022" role="doc-biblioref">2022</a>)</span>; <span class="citation" data-cites="hartiala2021">Hartiala et al. (<a href="09_references.html#ref-hartiala2021" role="doc-biblioref">2021</a>)</span>; <span class="citation" data-cites="van2018">Van Der Harst and Verweij (<a href="09_references.html#ref-van2018" role="doc-biblioref">2018</a>)</span>; <span class="citation" data-cites="aragam2022">Aragam et al. (<a href="09_references.html#ref-aragam2022" role="doc-biblioref">2022</a>)</span>; <span class="citation" data-cites="sudlow2015">Sudlow et al. (<a href="09_references.html#ref-sudlow2015" role="doc-biblioref">2015</a>)</span>; <span class="citation" data-cites="malik2018">Malik et al. (<a href="09_references.html#ref-malik2018" role="doc-biblioref">2018</a>)</span>; <span class="citation" data-cites="mishra2022">Mishra et al. (<a href="09_references.html#ref-mishra2022" role="doc-biblioref">2022</a>)</span>; <span class="citation" data-cites="malhotra2019">Malhotra et al. (<a href="09_references.html#ref-malhotra2019" role="doc-biblioref">2019</a>)</span>; <span class="citation" data-cites="morris2019">Morris et al. (<a href="09_references.html#ref-morris2019" role="doc-biblioref">2019</a>)</span></p>
</details>
</section>
</section>
<section id="data-processing" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="data-processing"><span class="header-section-number">2.2</span> Data processing</h2>
<section id="ld-matrix" class="level3" data-number="2.2.1">
<h3 data-number="2.2.1" class="anchored" data-anchor-id="ld-matrix"><span class="header-section-number">2.2.1</span> LD matrix</h3>
<p>The LD correlation matrix of the <em>SOST</em> region (100KB downstream and 125KB upstream of <em>SOST</em>) for the European samples from 1000 Genomes Phase 3 V5b <span class="citation" data-cites="genomes2015">(<a href="09_references.html#ref-genomes2015" role="doc-biblioref">1000 Genomes Project Consortium et al. 2015</a>)</span> is computed using the <code>01_data_ldmat.R</code> R script located in the <code>scripts</code> folder.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"./scripts/01_data_ldmat.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The output of the script is an <code>Rda</code> file saved in the <code>data</code> folder as <code>01_data_ld_mat.Rda</code> which contains two objects:</p>
<ul>
<li><code>ld_snp</code> - a <code>data.frame</code> of variant information with the columns:
<ul>
<li><code>rsid</code> - rsID<br>
</li>
<li><code>chr</code> - chromosome<br>
</li>
<li><code>pos</code> - position (build 37)<br>
</li>
<li><code>ref</code> - reference allele<br>
</li>
<li><code>alt</code> - alternate allele<br>
</li>
<li><code>af</code> - allele frequency of the alternate allele</li>
</ul></li>
<li><code>ld_mat</code> - a <code>matrix</code> of variant correlations (<span class="math inline">\(r\)</span> not <span class="math inline">\(r^2\)</span>)</li>
</ul>
</section>
<section id="gwas-data" class="level3" data-number="2.2.2">
<h3 data-number="2.2.2" class="anchored" data-anchor-id="gwas-data"><span class="header-section-number">2.2.2</span> GWAS data</h3>
<section id="sost-region" class="level4">
<h4 class="anchored" data-anchor-id="sost-region"><em>SOST</em> region</h4>
<p>The <em>SOST</em> region is extracted and harmonized from the GWAS datasets using the <code>02_data_gwas_sos_region.R</code> R script located in the <code>scripts</code> folder. The pQTL dataset is also harmonized using this script. The effect allele is aligned to the alternate allele in <code>ld_snps</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"./scripts/02_data_gwas_region.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The output of the script is an <code>Rda</code> file saved in the <code>data</code> folder as <code>02_data_gwas_sost_region.Rda</code> which contains three objects:</p>
<ul>
<li><code>gwas</code> - a <code>data.frame</code> of GWAS results from the studies above with the columns:
<ul>
<li><code>rsid</code> - rsID<br>
</li>
<li><code>chr</code> - chromosome<br>
</li>
<li><code>pos</code> - position (build 37)<br>
</li>
<li><code>ref</code> - reference allele<br>
</li>
<li><code>alt</code> - alternate allele (effect allele)<br>
</li>
<li><code>af</code> - allele frequency of the alternate allele<br>
</li>
<li><code>beta</code> - effect size<br>
</li>
<li><code>se</code> - standard error<br>
</li>
<li><code>pvalue</code> - <em>p</em>-value</li>
</ul></li>
<li><code>pqtls</code> - a <code>data.frame</code> of pQTL results with the columns:
<ul>
<li><code>rsid</code> - rsID<br>
</li>
<li><code>chr</code> - chromosome<br>
</li>
<li><code>pos</code> - position (build 37)<br>
</li>
<li><code>ref</code> - reference allele<br>
</li>
<li><code>alt</code> - alternate allele (effect allele)<br>
</li>
<li><code>af</code> - allele frequency of the alternate allele<br>
</li>
<li><code>beta</code> - effect size<br>
</li>
<li><code>se</code> - standard error<br>
</li>
<li><code>pvalue</code> - <em>p</em>-value</li>
</ul></li>
<li><code>studies</code> - a <code>data.frame</code> of GWAS study information with the columns:
<ul>
<li><code>id</code> - dataset ID<br>
</li>
<li><code>source</code> - source of dataset<br>
</li>
<li><code>pmid</code> - PubMed ID<br>
</li>
<li><code>author</code> - author of study<br>
</li>
<li><code>trait</code> - phenotype<br>
</li>
<li><code>abbr</code> - abbreviation<br>
</li>
<li><code>ancestry</code> - ancestry of study<br>
</li>
<li><code>n</code> - number of samples<br>
</li>
<li><code>n_cases</code> - number of cases<br>
</li>
<li><code>n_controls</code> - number of controls<br>
</li>
<li><code>unit</code> - unit of analysis (<code>IRNT</code> = inverse rank normal transformation)<br>
</li>
<li><code>flag</code> - flag if the dataset (or equivalent) was used by <span class="citation" data-cites="zheng2023">Zheng et al. (<a href="09_references.html#ref-zheng2023" role="doc-biblioref">2023</a>)</span> (<code>Y</code> = yes, <code>N</code> = no)</li>
</ul></li>
</ul>
</section>
<section id="pqtls" class="level4">
<h4 class="anchored" data-anchor-id="pqtls">pQTLs</h4>
<p>The GWAS results for the sclerostin pQTLs are extracted from <code>02_data_gwas_sost_region</code> using the <code>03_data_gwas_pqtls.R</code> R script located in the <code>scripts</code> folder. The associations of rs1107747 and rs4793023 with HDL cholesterol and triglycerides are adjusted for rs72836567 using the COJO methodology <span class="citation" data-cites="yang2012">(<a href="09_references.html#ref-yang2012" role="doc-biblioref">Yang et al. 2012</a>)</span> to account for the effects of the <em>CD300LG</em> gene (see <a href="03_ld_matrix.html#sec-ldmat"><span>Section&nbsp;3.2</span></a> for further details). No good proxy variants (<span class="math inline">\(r^2 \geq\)</span> 0.8) were identified for the missing sclerostin pQTLs in GCST006867.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"./scripts/03_data_gwas_pqtls.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The output of the script is an <code>Rda</code> file saved in the <code>data</code> folder as <code>03_data_gwas_pqtls.Rda</code> which contains three objects:</p>
<ul>
<li><code>gwas</code> - a <code>data.frame</code> of GWAS results from the studies above with the columns:
<ul>
<li><code>rsid</code> - rsID<br>
</li>
<li><code>chr</code> - chromosome<br>
</li>
<li><code>pos</code> - position (build 37)<br>
</li>
<li><code>ref</code> - reference allele<br>
</li>
<li><code>alt</code> - alternate allele (effect allele)<br>
</li>
<li><code>af</code> - allele frequency of the alternate allele<br>
</li>
<li><code>beta</code> - effect size<br>
</li>
<li><code>se</code> - standard error<br>
</li>
<li><code>pvalue</code> - <em>p</em>-value</li>
</ul></li>
<li><code>pqtls</code> - a <code>data.frame</code> of pQTL results with the columns:
<ul>
<li><code>rsid</code> - rsID<br>
</li>
<li><code>chr</code> - chromosome<br>
</li>
<li><code>pos</code> - position (build 37)<br>
</li>
<li><code>ref</code> - reference allele<br>
</li>
<li><code>alt</code> - alternate allele (effect allele)<br>
</li>
<li><code>af</code> - allele frequency of the alternate allele<br>
</li>
<li><code>beta</code> - effect size<br>
</li>
<li><code>se</code> - standard error<br>
</li>
<li><code>pvalue</code> - <em>p</em>-value</li>
</ul></li>
<li><code>studies</code> - a <code>data.frame</code> of GWAS study information with the columns:
<ul>
<li><code>id</code> - dataset ID<br>
</li>
<li><code>source</code> - source of dataset<br>
</li>
<li><code>pmid</code> - PubMed ID<br>
</li>
<li><code>author</code> - author of study<br>
</li>
<li><code>trait</code> - phenotype<br>
</li>
<li><code>abbr</code> - abbreviation<br>
</li>
<li><code>ancestry</code> - ancestry of study<br>
</li>
<li><code>n</code> - number of samples<br>
</li>
<li><code>n_cases</code> - number of cases<br>
</li>
<li><code>n_controls</code> - number of controls<br>
</li>
<li><code>unit</code> - unit of analysis (<code>IRNT</code> = inverse rank normal transformation)<br>
</li>
<li><code>flag</code> - flag if the dataset (or equivalent) was used by <span class="citation" data-cites="zheng2023">Zheng et al. (<a href="09_references.html#ref-zheng2023" role="doc-biblioref">2023</a>)</span> (<code>Y</code> = yes, <code>N</code> = no)</li>
</ul></li>
</ul>


<!-- -->

<div id="refs" class="references csl-bib-body hanging-indent" role="list" style="display: none">
<div id="ref-genomes2015" class="csl-entry" role="listitem">
1000 Genomes Project Consortium et al. A global reference for human genetic variation. Nature. 2015;526(7571):68.
</div>
<div id="ref-aragam2022" class="csl-entry" role="listitem">
Aragam KG, Jiang T, Goel A, et al. Discovery and systematic characterization of risk variants and genes for coronary artery disease in over a million participants. Nature Genetics. 2022;54(12):180315.
</div>
<div id="ref-hartiala2021" class="csl-entry" role="listitem">
Hartiala JA, Han Y, Jia Q, et al. Genome-wide analysis identifies novel susceptibility loci for myocardial infarction. European heart journal. 2021;42(9):91933.
</div>
<div id="ref-kavousi2022" class="csl-entry" role="listitem">
Kavousi M, Bos MM, Barnes HJ, et al. Multi-ancestry genome-wide analysis identifies effector genes and druggable pathways for coronary artery calcification. medRxiv. 2022;202205.
</div>
<div id="ref-mahajan2022" class="csl-entry" role="listitem">
Mahajan A, Spracklen CN, Zhang W, et al. Multi-ancestry genetic study of type 2 diabetes highlights the power of diverse populations for discovery and translation. Nature genetics. 2022;54(5):56072.
</div>
<div id="ref-malhotra2019" class="csl-entry" role="listitem">
Malhotra R, Mauer AC, Lino Cardenas CL, et al. HDAC9 is implicated in atherosclerotic aortic calcification and affects vascular smooth muscle cell phenotype. Nature genetics. 2019;51(11):15807.
</div>
<div id="ref-malik2018" class="csl-entry" role="listitem">
Malik R, Chauhan G, Traylor M, et al. Multiancestry genome-wide association study of 520,000 subjects identifies 32 loci associated with stroke and stroke subtypes. Nature genetics. 2018;50(4):52437.
</div>
<div id="ref-malik2016" class="csl-entry" role="listitem">
Malik R, Traylor M, Pulit SL, et al. Low-frequency and common genetic variation in ischemic stroke: The METASTROKE collaboration. Neurology. 2016;86(13):121726.
</div>
<div id="ref-mishra2022" class="csl-entry" role="listitem">
Mishra A, Malik R, Hachiya T, et al. Stroke genetics informs drug discovery and risk prediction across ancestries. Nature. 2022;611(7934):11523.
</div>
<div id="ref-morris2019" class="csl-entry" role="listitem">
Morris JA, Kemp JP, Youlten SE, et al. An atlas of genetic influences on osteoporosis in humans and mice. Nature genetics. 2019;51(2):25866.
</div>
<div id="ref-sudlow2015" class="csl-entry" role="listitem">
Sudlow C, Gallacher J, Allen N, et al. UK biobank: An open access resource for identifying the causes of a wide range of complex diseases of middle and old age. PLoS medicine. 2015;12(3):e1001779.
</div>
<div id="ref-van2018" class="csl-entry" role="listitem">
Van Der Harst P, Verweij N. Identification of 64 novel genetic loci provides an expanded view on the genetic architecture of coronary artery disease. Circulation research. 2018;122(3):43343.
</div>
<div id="ref-xue2018" class="csl-entry" role="listitem">
Xue A, Wu Y, Zhu Z, et al. Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes. Nature communications. 2018;9(1):2941.
</div>
<div id="ref-yang2012" class="csl-entry" role="listitem">
Yang J, Ferreira T, Morris AP, et al. Conditional and joint multiple-SNP analysis of GWAS summary statistics identifies additional variants influencing complex traits. Nature genetics. 2012;44(4):36975.
</div>
<div id="ref-zheng2023" class="csl-entry" role="listitem">
Zheng J, Wheeler E, Pietzner M, et al. Lowering of circulating sclerostin may increase risk of atherosclerosis and its risk factors: Evidence from a genome-wide association meta-analysis followed by mendelian randomization. Arthritis &amp; Rheumatology. 2023;75(10):178192.
</div>
</div>
</section>
</section>
</section>
<section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes">
<hr>
<ol>
<li id="fn1"><p>An extra 25KB was added upstream to ensure that all relevant <em>CD300LG</em> variants are included in the region.<a href="#fnref1" class="footnote-back" role="doc-backlink"></a></p></li>
<li id="fn2"><p>The ischemic and cardioembolic stroke GWAS results from METASTROKE <span class="citation" data-cites="malik2016">(<a href="09_references.html#ref-malik2016" role="doc-biblioref">Malik et al. 2016</a>)</span> used by <span class="citation" data-cites="zheng2023">Zheng et al. (<a href="09_references.html#ref-zheng2023" role="doc-biblioref">2023</a>)</span> were replaced with those from MEGASTROKE <span class="citation" data-cites="malik2018">(<a href="09_references.html#ref-malik2018" role="doc-biblioref">Malik et al. 2018</a>)</span> and the UK Biobank hypertension GWAS results from <a href="https://gwas.mrcieu.ac.uk/datasets/ukb-b-14057/">OpenGWAS</a> used by <span class="citation" data-cites="zheng2023">Zheng et al. (<a href="09_references.html#ref-zheng2023" role="doc-biblioref">2023</a>)</span> were replaced with those from <a href="https://pan.ukbb.broadinstitute.org/downloads">Pan-UKBB</a> due to licensing restrictions. The GWAS of coronary artery calcification was not available (either publicly or via application) at the time of this analysis <span class="citation" data-cites="kavousi2022">(<a href="09_references.html#ref-kavousi2022" role="doc-biblioref">Kavousi et al. 2022</a>)</span>.<a href="#fnref2" class="footnote-back" role="doc-backlink"></a></p></li>
<li id="fn3"><p>Since <span class="citation" data-cites="zheng2023">Zheng et al. (<a href="09_references.html#ref-zheng2023" role="doc-biblioref">2023</a>)</span> use trans-ethnic GWAS results, we followed suit. In all of the trans-ethnic analyses the majority of the samples were from European ancestry. There was no material difference in the results if we restricted the analyses to analysing Europeans only where possible <a href="08_sensitivity.html#sec-mr-eur"><span>Section&nbsp;8.1</span></a>.<a href="#fnref3" class="footnote-back" role="doc-backlink"></a></p></li>
</ol>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./index.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./03_ld_matrix.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">LD matrix</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb4" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>The sclerostin pQTLs and GWAS results for are processed using the scripts sourced below. The linkage disequilibrium (LD) matrix of genetic variants in the *SOST* region (100KB downstream and 125KB upstream[^1] of *SOST*) is computed using data from 1000 Genomes <span class="co">[</span><span class="ot">@genomes2015</span><span class="co">]</span>.  </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: setup</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Libraries</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reactable)</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Functions</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"./scripts/00_functions.R"</span>)</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a><span class="fu">## Datasets</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="fu">### Sclerostin pQTLs</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>The *cis* sclerostin pQTLs in <span class="co">[</span><span class="ot">@tbl-pqtls</span><span class="co">]</span> were extracted from Supplementary Table 2 in @zheng2023.</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: pqtls</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>pqtls <span class="ot">&lt;-</span> <span class="fu">fread</span>(</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>  <span class="st">"./data/00_data_pqtls.tsv"</span>,</span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">data.table =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-pqtls</span></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: cis pQTLs</span></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a><span class="fu">empty_kable</span>()</span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>pqtls <span class="sc">%&gt;%</span></span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reactable</span>(</span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">searchable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">pagination =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">bordered =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">striped =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a>      <span class="at">rsid =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"rsID"</span>, <span class="at">minWidth =</span> <span class="dv">100</span>, <span class="at">align =</span> <span class="st">"left"</span>),</span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>      <span class="at">chr =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Chr"</span>, <span class="at">minWidth =</span> <span class="dv">50</span>, <span class="at">align =</span> <span class="st">"left"</span>),</span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a>      <span class="at">pos =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Position"</span>, <span class="at">minWidth =</span> <span class="dv">90</span>, <span class="at">align =</span> <span class="st">"left"</span>),</span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a>      <span class="at">ea =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Effect allele"</span>, <span class="at">minWidth =</span> <span class="dv">80</span>, <span class="at">align =</span> <span class="st">"left"</span>),</span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a>      <span class="at">oa =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Other allele"</span>, <span class="at">minWidth =</span> <span class="dv">80</span>, <span class="at">align =</span> <span class="st">"left"</span>),</span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a>      <span class="at">eaf =</span> <span class="fu">colDef</span>(</span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Effect allele frequency"</span>, <span class="at">minWidth =</span> <span class="dv">100</span>, <span class="at">align =</span> <span class="st">"left"</span></span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a>      <span class="at">beta =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Effect size"</span>, <span class="at">minWidth =</span> <span class="dv">80</span>, <span class="at">align =</span> <span class="st">"left"</span>),</span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a>      <span class="at">se =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Standard error"</span>, <span class="at">minWidth =</span> <span class="dv">80</span>, <span class="at">align =</span> <span class="st">"left"</span>),</span>
<span id="cb4-66"><a href="#cb4-66" aria-hidden="true" tabindex="-1"></a>      <span class="at">pvalue =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"P-value"</span>, <span class="at">minWidth =</span> <span class="dv">80</span>, <span class="at">align =</span> <span class="st">"left"</span>),</span>
<span id="cb4-67"><a href="#cb4-67" aria-hidden="true" tabindex="-1"></a>      <span class="at">n =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Number of samples"</span>, <span class="at">minWidth =</span> <span class="dv">100</span>, <span class="at">align =</span> <span class="st">"left"</span>)</span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-69"><a href="#cb4-69" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb4-70"><a href="#cb4-70" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-71"><a href="#cb4-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-72"><a href="#cb4-72" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;br</span> <span class="er">style</span><span class="ot">=</span><span class="st">"line-height: 5px"</span> <span class="kw">/&gt;</span></span>
<span id="cb4-73"><a href="#cb4-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-76"><a href="#cb4-76" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-77"><a href="#cb4-77" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: download-pqtls</span></span>
<span id="cb4-78"><a href="#cb4-78" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb4-79"><a href="#cb4-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-80"><a href="#cb4-80" aria-hidden="true" tabindex="-1"></a>pqtls <span class="sc">%&gt;%</span></span>
<span id="cb4-81"><a href="#cb4-81" aria-hidden="true" tabindex="-1"></a>  downloadthis<span class="sc">::</span><span class="fu">download_this</span>(</span>
<span id="cb4-82"><a href="#cb4-82" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"pqtls"</span>,</span>
<span id="cb4-83"><a href="#cb4-83" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".xlsx"</span>,</span>
<span id="cb4-84"><a href="#cb4-84" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download as xlsx"</span>,</span>
<span id="cb4-85"><a href="#cb4-85" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"primary"</span>,</span>
<span id="cb4-86"><a href="#cb4-86" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-87"><a href="#cb4-87" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span></span>
<span id="cb4-88"><a href="#cb4-88" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb4-89"><a href="#cb4-89" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-90"><a href="#cb4-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-91"><a href="#cb4-91" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details&gt;</span></span>
<span id="cb4-92"><a href="#cb4-92" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>README<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb4-93"><a href="#cb4-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-94"><a href="#cb4-94" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`rsid`</span> - rsID  </span>
<span id="cb4-95"><a href="#cb4-95" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`chr`</span> - chromosome  </span>
<span id="cb4-96"><a href="#cb4-96" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`pos`</span> - position (build 37)  </span>
<span id="cb4-97"><a href="#cb4-97" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`ea`</span> - effect allele  </span>
<span id="cb4-98"><a href="#cb4-98" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`oa`</span> - other allele  </span>
<span id="cb4-99"><a href="#cb4-99" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`eaf`</span> - effect allele frequency  </span>
<span id="cb4-100"><a href="#cb4-100" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`beta`</span> - effect size</span>
<span id="cb4-101"><a href="#cb4-101" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`se`</span> - standard error of effect size</span>
<span id="cb4-102"><a href="#cb4-102" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`pvalue`</span> - *p*-pvalue</span>
<span id="cb4-103"><a href="#cb4-103" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`n`</span> - number of samples  </span>
<span id="cb4-104"><a href="#cb4-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-105"><a href="#cb4-105" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb4-106"><a href="#cb4-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-107"><a href="#cb4-107" aria-hidden="true" tabindex="-1"></a><span class="fu">### Study information</span></span>
<span id="cb4-108"><a href="#cb4-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-109"><a href="#cb4-109" aria-hidden="true" tabindex="-1"></a>The GWAS studies used in the analyses<span class="ot">[^2]</span> are presented in <span class="co">[</span><span class="ot">@tbl-gwas-study-info</span><span class="co">]</span>.<span class="ot">[^3]</span>  </span>
<span id="cb4-110"><a href="#cb4-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-113"><a href="#cb4-113" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-114"><a href="#cb4-114" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: gwas-study-info</span></span>
<span id="cb4-115"><a href="#cb4-115" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb4-116"><a href="#cb4-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-117"><a href="#cb4-117" aria-hidden="true" tabindex="-1"></a>studies <span class="ot">&lt;-</span> <span class="fu">fread</span>(</span>
<span id="cb4-118"><a href="#cb4-118" aria-hidden="true" tabindex="-1"></a>  <span class="st">"./data/00_data_studies.tsv"</span>,</span>
<span id="cb4-119"><a href="#cb4-119" aria-hidden="true" tabindex="-1"></a>  <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">data.table =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span></span>
<span id="cb4-120"><a href="#cb4-120" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-121"><a href="#cb4-121" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-122"><a href="#cb4-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-125"><a href="#cb4-125" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-126"><a href="#cb4-126" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-gwas-study-info</span></span>
<span id="cb4-127"><a href="#cb4-127" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: GWAS study information</span></span>
<span id="cb4-128"><a href="#cb4-128" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb4-129"><a href="#cb4-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-130"><a href="#cb4-130" aria-hidden="true" tabindex="-1"></a><span class="fu">empty_kable</span>()</span>
<span id="cb4-131"><a href="#cb4-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-132"><a href="#cb4-132" aria-hidden="true" tabindex="-1"></a>studies <span class="sc">%&gt;%</span></span>
<span id="cb4-133"><a href="#cb4-133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb4-134"><a href="#cb4-134" aria-hidden="true" tabindex="-1"></a>    <span class="at">id =</span> <span class="fu">if_else</span>(</span>
<span id="cb4-135"><a href="#cb4-135" aria-hidden="true" tabindex="-1"></a>      source <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"CHARGE"</span>, <span class="st">"DIAGRAM"</span>),</span>
<span id="cb4-136"><a href="#cb4-136" aria-hidden="true" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb4-137"><a href="#cb4-137" aria-hidden="true" tabindex="-1"></a>        <span class="st">"&lt;a href='"</span>,</span>
<span id="cb4-138"><a href="#cb4-138" aria-hidden="true" tabindex="-1"></a>        link,</span>
<span id="cb4-139"><a href="#cb4-139" aria-hidden="true" tabindex="-1"></a>        <span class="st">"'&gt;"</span>,</span>
<span id="cb4-140"><a href="#cb4-140" aria-hidden="true" tabindex="-1"></a>        id,</span>
<span id="cb4-141"><a href="#cb4-141" aria-hidden="true" tabindex="-1"></a>        <span class="st">"&lt;/a&gt;"</span></span>
<span id="cb4-142"><a href="#cb4-142" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb4-143"><a href="#cb4-143" aria-hidden="true" tabindex="-1"></a>      id</span>
<span id="cb4-144"><a href="#cb4-144" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb4-145"><a href="#cb4-145" aria-hidden="true" tabindex="-1"></a>    <span class="at">id =</span> <span class="fu">if_else</span>(</span>
<span id="cb4-146"><a href="#cb4-146" aria-hidden="true" tabindex="-1"></a>      source <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"GCST"</span>, <span class="st">"GEFOS"</span>, <span class="st">"MEGASTROKE"</span>),</span>
<span id="cb4-147"><a href="#cb4-147" aria-hidden="true" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb4-148"><a href="#cb4-148" aria-hidden="true" tabindex="-1"></a>        <span class="st">"&lt;a href='https://www.ebi.ac.uk/gwas/studies/"</span>,</span>
<span id="cb4-149"><a href="#cb4-149" aria-hidden="true" tabindex="-1"></a>        id,</span>
<span id="cb4-150"><a href="#cb4-150" aria-hidden="true" tabindex="-1"></a>        <span class="st">"/'&gt;"</span>,</span>
<span id="cb4-151"><a href="#cb4-151" aria-hidden="true" tabindex="-1"></a>        id,</span>
<span id="cb4-152"><a href="#cb4-152" aria-hidden="true" tabindex="-1"></a>        <span class="st">"&lt;/a&gt;"</span></span>
<span id="cb4-153"><a href="#cb4-153" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb4-154"><a href="#cb4-154" aria-hidden="true" tabindex="-1"></a>      id</span>
<span id="cb4-155"><a href="#cb4-155" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb4-156"><a href="#cb4-156" aria-hidden="true" tabindex="-1"></a>    <span class="at">id =</span> <span class="fu">if_else</span>(</span>
<span id="cb4-157"><a href="#cb4-157" aria-hidden="true" tabindex="-1"></a>      source <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"PUKB"</span>),</span>
<span id="cb4-158"><a href="#cb4-158" aria-hidden="true" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb4-159"><a href="#cb4-159" aria-hidden="true" tabindex="-1"></a>        <span class="st">"&lt;a href='https://pan.ukbb.broadinstitute.org/downloads'&gt;"</span>,</span>
<span id="cb4-160"><a href="#cb4-160" aria-hidden="true" tabindex="-1"></a>        id,</span>
<span id="cb4-161"><a href="#cb4-161" aria-hidden="true" tabindex="-1"></a>        <span class="st">"&lt;/a&gt;"</span></span>
<span id="cb4-162"><a href="#cb4-162" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb4-163"><a href="#cb4-163" aria-hidden="true" tabindex="-1"></a>      id</span>
<span id="cb4-164"><a href="#cb4-164" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb4-165"><a href="#cb4-165" aria-hidden="true" tabindex="-1"></a>    <span class="at">id =</span> <span class="fu">if_else</span>(</span>
<span id="cb4-166"><a href="#cb4-166" aria-hidden="true" tabindex="-1"></a>      source <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"UKB"</span>),</span>
<span id="cb4-167"><a href="#cb4-167" aria-hidden="true" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb4-168"><a href="#cb4-168" aria-hidden="true" tabindex="-1"></a>        <span class="st">"&lt;a href='http://www.nealelab.is/uk-biobank'&gt;"</span>,</span>
<span id="cb4-169"><a href="#cb4-169" aria-hidden="true" tabindex="-1"></a>        id,</span>
<span id="cb4-170"><a href="#cb4-170" aria-hidden="true" tabindex="-1"></a>        <span class="st">"&lt;/a&gt;"</span></span>
<span id="cb4-171"><a href="#cb4-171" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb4-172"><a href="#cb4-172" aria-hidden="true" tabindex="-1"></a>      id</span>
<span id="cb4-173"><a href="#cb4-173" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb4-174"><a href="#cb4-174" aria-hidden="true" tabindex="-1"></a>    <span class="at">pmid =</span> <span class="fu">paste0</span>(</span>
<span id="cb4-175"><a href="#cb4-175" aria-hidden="true" tabindex="-1"></a>      <span class="st">"&lt;a href='https://pubmed.ncbi.nlm.nih.gov/"</span>,</span>
<span id="cb4-176"><a href="#cb4-176" aria-hidden="true" tabindex="-1"></a>      pmid,</span>
<span id="cb4-177"><a href="#cb4-177" aria-hidden="true" tabindex="-1"></a>      <span class="st">"'&gt;"</span>,</span>
<span id="cb4-178"><a href="#cb4-178" aria-hidden="true" tabindex="-1"></a>      pmid,</span>
<span id="cb4-179"><a href="#cb4-179" aria-hidden="true" tabindex="-1"></a>      <span class="st">"&lt;/a&gt;"</span></span>
<span id="cb4-180"><a href="#cb4-180" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-181"><a href="#cb4-181" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb4-182"><a href="#cb4-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(id, pmid, trait, abbr, ancestry, n, n_cases, n_controls) <span class="sc">%&gt;%</span></span>
<span id="cb4-183"><a href="#cb4-183" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reactable</span>(</span>
<span id="cb4-184"><a href="#cb4-184" aria-hidden="true" tabindex="-1"></a>    <span class="at">searchable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-185"><a href="#cb4-185" aria-hidden="true" tabindex="-1"></a>    <span class="at">pagination =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-186"><a href="#cb4-186" aria-hidden="true" tabindex="-1"></a>    <span class="at">bordered =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-187"><a href="#cb4-187" aria-hidden="true" tabindex="-1"></a>    <span class="at">striped =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-188"><a href="#cb4-188" aria-hidden="true" tabindex="-1"></a>    <span class="at">columns =</span> <span class="fu">list</span>(</span>
<span id="cb4-189"><a href="#cb4-189" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> <span class="fu">colDef</span>(</span>
<span id="cb4-190"><a href="#cb4-190" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Dataset ID"</span>, <span class="at">minWidth =</span> <span class="dv">100</span>, <span class="at">align =</span> <span class="st">"left"</span>, <span class="at">html =</span> <span class="cn">TRUE</span></span>
<span id="cb4-191"><a href="#cb4-191" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb4-192"><a href="#cb4-192" aria-hidden="true" tabindex="-1"></a>      <span class="at">pmid =</span> <span class="fu">colDef</span>(</span>
<span id="cb4-193"><a href="#cb4-193" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"PubMed ID"</span>, <span class="at">minWidth =</span> <span class="dv">80</span>, <span class="at">align =</span> <span class="st">"left"</span>, <span class="at">html =</span> <span class="cn">TRUE</span></span>
<span id="cb4-194"><a href="#cb4-194" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb4-195"><a href="#cb4-195" aria-hidden="true" tabindex="-1"></a>      <span class="at">trait =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Phenotype"</span>, <span class="at">minWidth =</span> <span class="dv">140</span>, <span class="at">align =</span> <span class="st">"left"</span>),</span>
<span id="cb4-196"><a href="#cb4-196" aria-hidden="true" tabindex="-1"></a>      <span class="at">abbr =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Abbreviation"</span>, <span class="at">minWidth =</span> <span class="dv">100</span>, <span class="at">align =</span> <span class="st">"left"</span>),</span>
<span id="cb4-197"><a href="#cb4-197" aria-hidden="true" tabindex="-1"></a>      <span class="at">ancestry =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Ancestry"</span>, <span class="at">minWidth =</span> <span class="dv">100</span>, <span class="at">align =</span> <span class="st">"left"</span>),</span>
<span id="cb4-198"><a href="#cb4-198" aria-hidden="true" tabindex="-1"></a>      <span class="at">n =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Number of samples"</span>, <span class="at">minWidth =</span> <span class="dv">80</span>, <span class="at">align =</span> <span class="st">"left"</span>),</span>
<span id="cb4-199"><a href="#cb4-199" aria-hidden="true" tabindex="-1"></a>      <span class="at">n_cases =</span> <span class="fu">colDef</span>(<span class="at">name =</span> <span class="st">"Number of cases"</span>, <span class="at">minWidth =</span> <span class="dv">80</span>, <span class="at">align =</span> <span class="st">"left"</span>),</span>
<span id="cb4-200"><a href="#cb4-200" aria-hidden="true" tabindex="-1"></a>      <span class="at">n_controls =</span> <span class="fu">colDef</span>(</span>
<span id="cb4-201"><a href="#cb4-201" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Number of controls"</span>, <span class="at">minWidth =</span> <span class="dv">80</span>, <span class="at">align =</span> <span class="st">"left"</span></span>
<span id="cb4-202"><a href="#cb4-202" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb4-203"><a href="#cb4-203" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-204"><a href="#cb4-204" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb4-205"><a href="#cb4-205" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-206"><a href="#cb4-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-209"><a href="#cb4-209" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-210"><a href="#cb4-210" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: download-gwas-studies</span></span>
<span id="cb4-211"><a href="#cb4-211" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb4-212"><a href="#cb4-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-213"><a href="#cb4-213" aria-hidden="true" tabindex="-1"></a>studies <span class="sc">%&gt;%</span></span>
<span id="cb4-214"><a href="#cb4-214" aria-hidden="true" tabindex="-1"></a>  downloadthis<span class="sc">::</span><span class="fu">download_this</span>(</span>
<span id="cb4-215"><a href="#cb4-215" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_name =</span> <span class="st">"studies"</span>,</span>
<span id="cb4-216"><a href="#cb4-216" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_extension =</span> <span class="st">".xlsx"</span>,</span>
<span id="cb4-217"><a href="#cb4-217" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_label =</span> <span class="st">"Download as xlsx"</span>,</span>
<span id="cb4-218"><a href="#cb4-218" aria-hidden="true" tabindex="-1"></a>    <span class="at">button_type =</span> <span class="st">"primary"</span>,</span>
<span id="cb4-219"><a href="#cb4-219" aria-hidden="true" tabindex="-1"></a>    <span class="at">has_icon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-220"><a href="#cb4-220" aria-hidden="true" tabindex="-1"></a>    <span class="at">icon =</span> <span class="st">"fa fa-save"</span></span>
<span id="cb4-221"><a href="#cb4-221" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb4-222"><a href="#cb4-222" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-223"><a href="#cb4-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-224"><a href="#cb4-224" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;details&gt;</span></span>
<span id="cb4-225"><a href="#cb4-225" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;summary&gt;</span>README<span class="kw">&lt;/summary&gt;</span></span>
<span id="cb4-226"><a href="#cb4-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-227"><a href="#cb4-227" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`id`</span> - dataset ID  </span>
<span id="cb4-228"><a href="#cb4-228" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`source`</span> - source of dataset  </span>
<span id="cb4-229"><a href="#cb4-229" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`pmid`</span> - PubMed ID  </span>
<span id="cb4-230"><a href="#cb4-230" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`author`</span> - author of study  </span>
<span id="cb4-231"><a href="#cb4-231" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`link`</span> - link to dataset  </span>
<span id="cb4-232"><a href="#cb4-232" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`trait`</span> - phenotype  </span>
<span id="cb4-233"><a href="#cb4-233" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`abbr`</span> - abbreviation  </span>
<span id="cb4-234"><a href="#cb4-234" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`ancestry`</span> - ancestry of study  </span>
<span id="cb4-235"><a href="#cb4-235" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`n`</span> - number of samples  </span>
<span id="cb4-236"><a href="#cb4-236" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`n_cases`</span> - number of cases  </span>
<span id="cb4-237"><a href="#cb4-237" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`n_controls`</span> - number of controls  </span>
<span id="cb4-238"><a href="#cb4-238" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`unit`</span> - unit of analysis (<span class="in">`IRNT`</span> = inverse rank normal transformation)  </span>
<span id="cb4-239"><a href="#cb4-239" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`flag`</span> - flag if the dataset (or equivalent) was used by @zheng2023 (<span class="in">`Y`</span> = yes, <span class="in">`N`</span> = no)  </span>
<span id="cb4-240"><a href="#cb4-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-241"><a href="#cb4-241" aria-hidden="true" tabindex="-1"></a>Citations: @xue2018; @mahajan2022; @hartiala2021; @van2018; @aragam2022; @sudlow2015; @malik2018; @mishra2022; @malhotra2019; @morris2019  </span>
<span id="cb4-242"><a href="#cb4-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-243"><a href="#cb4-243" aria-hidden="true" tabindex="-1"></a><span class="kw">&lt;/details&gt;</span></span>
<span id="cb4-244"><a href="#cb4-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-245"><a href="#cb4-245" aria-hidden="true" tabindex="-1"></a><span class="fu">## Data processing</span></span>
<span id="cb4-246"><a href="#cb4-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-247"><a href="#cb4-247" aria-hidden="true" tabindex="-1"></a><span class="fu">### LD matrix</span></span>
<span id="cb4-248"><a href="#cb4-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-249"><a href="#cb4-249" aria-hidden="true" tabindex="-1"></a>The LD correlation matrix of the *SOST* region (100KB downstream and 125KB upstream of *SOST*) for the European samples from 1000 Genomes Phase 3 V5b <span class="co">[</span><span class="ot">@genomes2015</span><span class="co">]</span> is computed using the <span class="in">`01_data_ldmat.R`</span> R script located in the <span class="in">`scripts`</span> folder.  </span>
<span id="cb4-250"><a href="#cb4-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-253"><a href="#cb4-253" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-254"><a href="#cb4-254" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: data-ld</span></span>
<span id="cb4-255"><a href="#cb4-255" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb4-256"><a href="#cb4-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-257"><a href="#cb4-257" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"./scripts/01_data_ldmat.R"</span>)</span>
<span id="cb4-258"><a href="#cb4-258" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-259"><a href="#cb4-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-260"><a href="#cb4-260" aria-hidden="true" tabindex="-1"></a>The output of the script is an <span class="in">`Rda`</span> file saved in the <span class="in">`data`</span> folder as <span class="in">`01_data_ld_mat.Rda`</span> which contains two objects:  </span>
<span id="cb4-261"><a href="#cb4-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-262"><a href="#cb4-262" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`ld_snp`</span> - a <span class="in">`data.frame`</span> of variant information with the columns:  </span>
<span id="cb4-263"><a href="#cb4-263" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`rsid`</span> - rsID  </span>
<span id="cb4-264"><a href="#cb4-264" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`chr`</span> - chromosome  </span>
<span id="cb4-265"><a href="#cb4-265" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`pos`</span> - position (build 37)  </span>
<span id="cb4-266"><a href="#cb4-266" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`ref`</span> - reference allele  </span>
<span id="cb4-267"><a href="#cb4-267" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`alt`</span> - alternate allele  </span>
<span id="cb4-268"><a href="#cb4-268" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`af`</span> - allele frequency of the alternate allele  </span>
<span id="cb4-269"><a href="#cb4-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-270"><a href="#cb4-270" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`ld_mat`</span> - a <span class="in">`matrix`</span> of variant correlations ($r$ not $r^2$)  </span>
<span id="cb4-271"><a href="#cb4-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-272"><a href="#cb4-272" aria-hidden="true" tabindex="-1"></a><span class="fu">### GWAS data</span></span>
<span id="cb4-273"><a href="#cb4-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-274"><a href="#cb4-274" aria-hidden="true" tabindex="-1"></a><span class="fu">#### *SOST* region</span></span>
<span id="cb4-275"><a href="#cb4-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-276"><a href="#cb4-276" aria-hidden="true" tabindex="-1"></a>The *SOST* region is extracted and harmonized from the GWAS datasets using the <span class="in">`02_data_gwas_sos_region.R`</span> R script located in the <span class="in">`scripts`</span> folder. The pQTL dataset is also harmonized using this script. The effect allele is aligned to the alternate allele in <span class="in">`ld_snps`</span>.  </span>
<span id="cb4-277"><a href="#cb4-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-280"><a href="#cb4-280" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-281"><a href="#cb4-281" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: data-gwas-region</span></span>
<span id="cb4-282"><a href="#cb4-282" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb4-283"><a href="#cb4-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-284"><a href="#cb4-284" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"./scripts/02_data_gwas_region.R"</span>)</span>
<span id="cb4-285"><a href="#cb4-285" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-286"><a href="#cb4-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-287"><a href="#cb4-287" aria-hidden="true" tabindex="-1"></a>The output of the script is an <span class="in">`Rda`</span> file saved in the <span class="in">`data`</span> folder as <span class="in">`02_data_gwas_sost_region.Rda`</span> which contains three objects:  </span>
<span id="cb4-288"><a href="#cb4-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-289"><a href="#cb4-289" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`gwas`</span> - a <span class="in">`data.frame`</span> of GWAS results from the studies above with the columns:  </span>
<span id="cb4-290"><a href="#cb4-290" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`rsid`</span> - rsID  </span>
<span id="cb4-291"><a href="#cb4-291" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`chr`</span> - chromosome  </span>
<span id="cb4-292"><a href="#cb4-292" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`pos`</span> - position (build 37)  </span>
<span id="cb4-293"><a href="#cb4-293" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`ref`</span> - reference allele  </span>
<span id="cb4-294"><a href="#cb4-294" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`alt`</span> - alternate allele (effect allele)  </span>
<span id="cb4-295"><a href="#cb4-295" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`af`</span> - allele frequency of the alternate allele  </span>
<span id="cb4-296"><a href="#cb4-296" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`beta`</span> - effect size  </span>
<span id="cb4-297"><a href="#cb4-297" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`se`</span> - standard error  </span>
<span id="cb4-298"><a href="#cb4-298" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`pvalue`</span> - *p*-value  </span>
<span id="cb4-299"><a href="#cb4-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-300"><a href="#cb4-300" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`pqtls`</span> - a <span class="in">`data.frame`</span> of pQTL results with the columns:  </span>
<span id="cb4-301"><a href="#cb4-301" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`rsid`</span> - rsID  </span>
<span id="cb4-302"><a href="#cb4-302" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`chr`</span> - chromosome  </span>
<span id="cb4-303"><a href="#cb4-303" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`pos`</span> - position (build 37)  </span>
<span id="cb4-304"><a href="#cb4-304" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`ref`</span> - reference allele  </span>
<span id="cb4-305"><a href="#cb4-305" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`alt`</span> - alternate allele (effect allele)  </span>
<span id="cb4-306"><a href="#cb4-306" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`af`</span> - allele frequency of the alternate allele  </span>
<span id="cb4-307"><a href="#cb4-307" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`beta`</span> - effect size  </span>
<span id="cb4-308"><a href="#cb4-308" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`se`</span> - standard error  </span>
<span id="cb4-309"><a href="#cb4-309" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`pvalue`</span> - *p*-value  </span>
<span id="cb4-310"><a href="#cb4-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-311"><a href="#cb4-311" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`studies`</span> - a <span class="in">`data.frame`</span> of GWAS study information with the columns:  </span>
<span id="cb4-312"><a href="#cb4-312" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`id`</span> - dataset ID  </span>
<span id="cb4-313"><a href="#cb4-313" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`source`</span> - source of dataset  </span>
<span id="cb4-314"><a href="#cb4-314" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`pmid`</span> - PubMed ID  </span>
<span id="cb4-315"><a href="#cb4-315" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`author`</span> - author of study  </span>
<span id="cb4-316"><a href="#cb4-316" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`trait`</span> - phenotype  </span>
<span id="cb4-317"><a href="#cb4-317" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`abbr`</span> - abbreviation  </span>
<span id="cb4-318"><a href="#cb4-318" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`ancestry`</span> - ancestry of study  </span>
<span id="cb4-319"><a href="#cb4-319" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`n`</span> - number of samples  </span>
<span id="cb4-320"><a href="#cb4-320" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`n_cases`</span> - number of cases  </span>
<span id="cb4-321"><a href="#cb4-321" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`n_controls`</span> - number of controls  </span>
<span id="cb4-322"><a href="#cb4-322" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`unit`</span> - unit of analysis (<span class="in">`IRNT`</span> = inverse rank normal transformation)  </span>
<span id="cb4-323"><a href="#cb4-323" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`flag`</span> - flag if the dataset (or equivalent) was used by @zheng2023 (<span class="in">`Y`</span> = yes, <span class="in">`N`</span> = no)  </span>
<span id="cb4-324"><a href="#cb4-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-325"><a href="#cb4-325" aria-hidden="true" tabindex="-1"></a><span class="fu">#### pQTLs</span></span>
<span id="cb4-326"><a href="#cb4-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-327"><a href="#cb4-327" aria-hidden="true" tabindex="-1"></a>The GWAS results for the sclerostin pQTLs are extracted from <span class="in">`02_data_gwas_sost_region`</span> using the <span class="in">`03_data_gwas_pqtls.R`</span> R script located in the <span class="in">`scripts`</span> folder. The associations of rs1107747 and rs4793023 with HDL cholesterol and triglycerides are adjusted for rs72836567 using the COJO methodology <span class="co">[</span><span class="ot">@yang2012</span><span class="co">]</span> to account for the effects of the *CD300LG* gene (see <span class="co">[</span><span class="ot">@sec-ldmat</span><span class="co">]</span> for further details). No good proxy variants ($r^2 \geq$ 0.8) were identified for the missing sclerostin pQTLs in GCST006867.  </span>
<span id="cb4-328"><a href="#cb4-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-331"><a href="#cb4-331" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-332"><a href="#cb4-332" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: data-gwas-pqtls</span></span>
<span id="cb4-333"><a href="#cb4-333" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb4-334"><a href="#cb4-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-335"><a href="#cb4-335" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"./scripts/03_data_gwas_pqtls.R"</span>)</span>
<span id="cb4-336"><a href="#cb4-336" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-337"><a href="#cb4-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-338"><a href="#cb4-338" aria-hidden="true" tabindex="-1"></a>The output of the script is an <span class="in">`Rda`</span> file saved in the <span class="in">`data`</span> folder as <span class="in">`03_data_gwas_pqtls.Rda`</span> which contains three objects:  </span>
<span id="cb4-339"><a href="#cb4-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-340"><a href="#cb4-340" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`gwas`</span> - a <span class="in">`data.frame`</span> of GWAS results from the studies above with the columns:  </span>
<span id="cb4-341"><a href="#cb4-341" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`rsid`</span> - rsID  </span>
<span id="cb4-342"><a href="#cb4-342" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`chr`</span> - chromosome  </span>
<span id="cb4-343"><a href="#cb4-343" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`pos`</span> - position (build 37)  </span>
<span id="cb4-344"><a href="#cb4-344" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`ref`</span> - reference allele  </span>
<span id="cb4-345"><a href="#cb4-345" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`alt`</span> - alternate allele (effect allele)  </span>
<span id="cb4-346"><a href="#cb4-346" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`af`</span> - allele frequency of the alternate allele  </span>
<span id="cb4-347"><a href="#cb4-347" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`beta`</span> - effect size  </span>
<span id="cb4-348"><a href="#cb4-348" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`se`</span> - standard error  </span>
<span id="cb4-349"><a href="#cb4-349" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`pvalue`</span> - *p*-value  </span>
<span id="cb4-350"><a href="#cb4-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-351"><a href="#cb4-351" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`pqtls`</span> - a <span class="in">`data.frame`</span> of pQTL results with the columns:  </span>
<span id="cb4-352"><a href="#cb4-352" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`rsid`</span> - rsID  </span>
<span id="cb4-353"><a href="#cb4-353" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`chr`</span> - chromosome  </span>
<span id="cb4-354"><a href="#cb4-354" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`pos`</span> - position (build 37)  </span>
<span id="cb4-355"><a href="#cb4-355" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`ref`</span> - reference allele  </span>
<span id="cb4-356"><a href="#cb4-356" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`alt`</span> - alternate allele (effect allele)  </span>
<span id="cb4-357"><a href="#cb4-357" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`af`</span> - allele frequency of the alternate allele  </span>
<span id="cb4-358"><a href="#cb4-358" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`beta`</span> - effect size  </span>
<span id="cb4-359"><a href="#cb4-359" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`se`</span> - standard error  </span>
<span id="cb4-360"><a href="#cb4-360" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`pvalue`</span> - *p*-value  </span>
<span id="cb4-361"><a href="#cb4-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-362"><a href="#cb4-362" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`studies`</span> - a <span class="in">`data.frame`</span> of GWAS study information with the columns:  </span>
<span id="cb4-363"><a href="#cb4-363" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`id`</span> - dataset ID  </span>
<span id="cb4-364"><a href="#cb4-364" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`source`</span> - source of dataset  </span>
<span id="cb4-365"><a href="#cb4-365" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`pmid`</span> - PubMed ID  </span>
<span id="cb4-366"><a href="#cb4-366" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`author`</span> - author of study  </span>
<span id="cb4-367"><a href="#cb4-367" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`trait`</span> - phenotype  </span>
<span id="cb4-368"><a href="#cb4-368" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`abbr`</span> - abbreviation  </span>
<span id="cb4-369"><a href="#cb4-369" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`ancestry`</span> - ancestry of study  </span>
<span id="cb4-370"><a href="#cb4-370" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`n`</span> - number of samples  </span>
<span id="cb4-371"><a href="#cb4-371" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`n_cases`</span> - number of cases  </span>
<span id="cb4-372"><a href="#cb4-372" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`n_controls`</span> - number of controls  </span>
<span id="cb4-373"><a href="#cb4-373" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`unit`</span> - unit of analysis (<span class="in">`IRNT`</span> = inverse rank normal transformation)  </span>
<span id="cb4-374"><a href="#cb4-374" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`flag`</span> - flag if the dataset (or equivalent) was used by @zheng2023 (<span class="in">`Y`</span> = yes, <span class="in">`N`</span> = no)  </span>
<span id="cb4-375"><a href="#cb4-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-376"><a href="#cb4-376" aria-hidden="true" tabindex="-1"></a><span class="ot">[^1]: </span>An extra 25KB was added upstream to ensure that all relevant *CD300LG* variants are included in the region.  </span>
<span id="cb4-377"><a href="#cb4-377" aria-hidden="true" tabindex="-1"></a><span class="ot">[^2]: </span>The ischemic and cardioembolic stroke GWAS results from METASTROKE <span class="co">[</span><span class="ot">@malik2016</span><span class="co">]</span> used by @zheng2023 were replaced with those from MEGASTROKE <span class="co">[</span><span class="ot">@malik2018</span><span class="co">]</span> and the UK Biobank hypertension GWAS results from <span class="co">[</span><span class="ot">OpenGWAS</span><span class="co">](https://gwas.mrcieu.ac.uk/datasets/ukb-b-14057/)</span> used by @zheng2023 were replaced with those from <span class="co">[</span><span class="ot">Pan-UKBB</span><span class="co">](https://pan.ukbb.broadinstitute.org/downloads)</span> due to licensing restrictions. The GWAS of coronary artery calcification was not available (either publicly or via application) at the time of this analysis <span class="co">[</span><span class="ot">@kavousi2022</span><span class="co">]</span>.  </span>
<span id="cb4-378"><a href="#cb4-378" aria-hidden="true" tabindex="-1"></a><span class="ot">[^3]: </span>Since @zheng2023 use trans-ethnic GWAS results, we followed suit. In all of the trans-ethnic analyses the majority of the samples were from European ancestry. There was no material difference in the results if we restricted the analyses to analysing Europeans only where possible <span class="co">[</span><span class="ot">@sec-mr-eur</span><span class="co">]</span>.  </span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left"><a href="https://www.gnu.org/licenses/gpl-3.0.en.html"><img src="https://img.shields.io/badge/Code-GPL--3-BD0000.svg" class="img-fluid"></a> <a href="https://creativecommons.org/licenses/by/4.0/"><img src="https://img.shields.io/badge/Data-CC--BY--4.0-00AFD4.svg" class="img-fluid"></a></div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">James Staley</div>
  </div>
</footer>



</body></html>